---
title: "01-dge"
author: "Sarah Tanja"
date: "4/4/2023"
output: html_document
---

# Differential Gene Expression

<https://sr320.github.io/course-fish546-2023/assignments/02-DGE.html>

For this assignment you will be taking RNA-seq reads off the sequencer, and determining what genes are expressed higher in [treatment group A]{.underline} compared to [treatments group B]{.underline}. The end product will be a plot and table of deferentially expressed genes.

> Kallisto is a pseudo-aligner, it does not need a genome to align to!

## Setup git hooks to automatically gitignore files \>100MB

You can use Git's built-in hooks to automatically ignore files larger than 100 MB. Here are the steps to follow:

-   Create a new text file in the .git/hooks/ directory of your repository called pre-commit. *Select the `More` tab with the gear icon under the RStudio Files navigator bar and select 'show hidden files' to see the .git folder*.

-   Add the following code to the pre-commit file:

```{txt}
#!/bin/bash

# Maximum file size (in bytes)
max_file_size=104857600

# Find all files larger than max_file_size and add them to the .gitignore file
find . -type f -size +$max_file_sizec -exec echo "{}" >> .gitignore \;
```

This code sets the max_file_size variable to 100 MB and then uses the find command to locate all files in the repository that are larger than the specified max_file_size. The exec option of the find command appends the name of each file that matches the criteria to the .gitignore file.

Save the pre-commit file and make it executable by running the following command in Terminal:

```{bash}
stanja@raven:~/sarahtanja/sarahtanja-coursework$ chmod +x .git/hooks/pre-commit
```

With these changes, whenever you run a git commit command, Git will first execute the pre-commit hook, which will automatically add any files larger than 100 MB to the .gitignore file. This will prevent Git from tracking these files in the repository going forward.

> In the event that you accidentally committed a file \>100MB, you can reset to the last successful git master branch push by following [this](https://stackoverflow.com/questions/1125968/how-do-i-force-git-pull-to-overwrite-local-files) stackoverflow answer:
>
> First, update all `origin/<branch>` refs to latest:
>
> ```{bash}
> git fetch --all
> ```
>
> Backup your current branch (e.g. `master`):
>
> ```{bash}
> git branch backup-master
> ```
>
> Jump to the latest commit on `origin/master` and checkout those files:
>
> ```{bash}
> git reset --hard origin/master
> ```

## Step 1. Download fastq.gz files

Note: add assignments/code/\*fastq to .gitignore file! These files, even zipped, are greater than 100MB and cannot be committed to Github.

Checkout the readme.md

```{bash}
# output readme file to the terminal 
curl https://owl.fish.washington.edu/nightingales/C_gigas/readme.md
```

Download all the fastq.gz files (this takes a long time!)

```{bash}
# navigate to new file in data folder
cd ../data/02-dge
# download fastq.gz files using wget 
wget --recursive --no-parent --no-directories --no-check-certificate\
--accept '[DMN]\*001.fastq.gz'\
https://owl.fish.washington.edu/nightingales/C_gigas/
```

## Step 2. Install Kallisto

For this assignment we will be using kallisto (bash), DESeq2 (r).

Create a directory in your home directory (outside of git repo) named `programs`, then use `wget` to download zipped program software into the `programs` directory

```{bash}
pwd
cd ../../../../
mkdir programs
cd programs 
wget https://github.com/pachterlab/kallisto/releases/download/v0.46.1/kallisto_linux-v0.46.1.tar.gz
```

Unzip the file

```{bash}
gunzip ~/sarahtanja/programs/kallisto_linux-v0.46.1.tar.gz
```

## Step 3. Running kallisto

Kallisto is already installed on raven (`/home/shared/kallisto/kallisto`) \> Note: Cannot find `shared` folder under /home in raven... 04APR23

```{bash}
/home/shared/kallisto/kallisto index
```

## Step 4. Downloading Reference

```{bash}
# change to work in data directory
cd ../data
# download fna file to data directory
wget --no-check-certificate https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/rna.fna
```

creating an index
```{bash}
# use absolute path to kallisto program
 /home/shared/kallisto/kallisto \
# create an index and name it (-i) `cgigas_roslin_rna.index`
index -i \
../data/cgigas_roslin_rna.index \
# here is the absolute path to the RNAseq data (fasta files)
/home/shared/8TB_HDD_01/sr320/github/nb-2023/Cgigas/data
```

## Step 5. Download sequence reads
data is located at absolute path `/home/shared/8TB_HDD_01/sr320/github/nb-2023/Cgigas/data`

```{bash}
# make kallisto_01 directory in output directorty
# comment out once directory is made

#cd ../output
#mkdir kallisto_01
```


```{bash}
# replace data location with absolute path above
#find ../data/*fastq.gz \ # take the data

find /home/shared/8TB_HDD_01/sr320/github/nb-2023/Cgigas/data \
| xargs basename -s _R1_001.fastq.gz | xargs -I{} /home/shared/kallisto/kallisto \
quant -i ../data/cgigas_roslin_rna.index \
-o ../output/kallisto_01/{} \
-t 40 \
--single -l 100 -s 10 ../data/{}_L001_R1_001.fastq.gz

```
