<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.335">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Sarah Tanja">
<meta name="dcterms.date" content="2023-04-13">

<title>Differential Gene Expression Analysis with Kallisto &amp; DESeq2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { color: #008000; } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { color: #008000; font-weight: bold; } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>


<script src="03-dge-pretty_files/libs/clipboard/clipboard.min.js"></script>
<script src="03-dge-pretty_files/libs/quarto-html/quarto.js"></script>
<script src="03-dge-pretty_files/libs/quarto-html/popper.min.js"></script>
<script src="03-dge-pretty_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="03-dge-pretty_files/libs/quarto-html/anchor.min.js"></script>
<link href="03-dge-pretty_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="03-dge-pretty_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="03-dge-pretty_files/libs/quarto-html/quarto-syntax-highlighting-dark.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<script src="03-dge-pretty_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="03-dge-pretty_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="03-dge-pretty_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="03-dge-pretty_files/libs/bootstrap/bootstrap-dark.min.css" rel="prefetch" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Differential Gene Expression Analysis with Kallisto &amp; DESeq2</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
<p class="subtitle lead">a pretty report for <a href="https://sr320.github.io/course-fish546-2023/assignments/03-knit.html">FISH 546</a></p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Sarah Tanja </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">April 13, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<section id="advanced-prep" class="level1">
<h1>Advanced Prep</h1>
<p>For this assignment you will be taking RNA-seq data, and running a common differential gene expression analysis, without the use of a reference genome. The end product will be a number of plots and table of differentially expressed genes.</p>
<div class="callout-important callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Important
</div>
</div>
<div class="callout-body-container callout-body">
<p><strong>Pre-reqs:</strong> This code contains relative paths that only work within the <em>raven</em> Roberts’ Lab Server, where the data for this assignment is housed</p>
</div>
</div>
<hr>
<p>Install Packages</p>
<div class="cell" data-layout-align="center" data-hash="03-dge-pretty_cache/html/packages_3556b7ffc435b66b2cfb50344e790f5a">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"knitr"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"knitr"</span>) </span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"tidyverse"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"kableExtra"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'kableExtra'</span>)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"pheatmap"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'pheatmap'</span>) </span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"RColorBrewer"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'RColorBrewer'</span>) </span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"data.table"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'data.table'</span>) </span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"DT"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'DT'</span>)</span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"Biostrings"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'Biostrings'</span>) </span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"BiocManager"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'BiocManager'</span>) </span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"DESeq2"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'DESeq2'</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Load Libraries</p>
<div class="cell" data-layout-align="center" data-hash="03-dge-pretty_cache/html/libraries_60831135d0e02ccc159569afc935bdab">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb2-9"><a href="#cb2-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
<section id="setup-git-hooks-to-automatically-gitignore-large-files" class="level2">
<h2 class="anchored" data-anchor-id="setup-git-hooks-to-automatically-gitignore-large-files">Setup git hooks to automatically gitignore large files</h2>
<p>In this assignment we’re dealing with large files that will end up in the ../data and ../output directories. To prevent those large files from clogging up our ability to ‘git push’, we can use built-in hooks to automatically ignore files larger than 100 MB (no matter the directory or file name!). Here are the steps to follow:</p>
<ul>
<li><p>Create a new text file in the <code>.git/hooks/</code> directory of your repository called <code>pre-commit</code></p>
<blockquote class="blockquote">
<p><em>Select the <code>More</code> tab with the gear icon under the RStudio Files navigator bar and select ‘show hidden files’ to see the .git folder</em>.</p>
</blockquote></li>
<li><p>Add the following text to the <code>.git/hooks/pre-commit</code> file:</p></li>
</ul>
<pre><code>#!/bin/sh

# Maximum file size (in bytes)
max_file_size=104857600

# Find all files larger than max_file_size and add them to the .gitignore file
find . -type f -size +$max_file_size -exec echo "{}" &gt;&gt; .gitignore \;</code></pre>
<p>This code sets the max_file_size variable to 100 MB and then uses the find command to locate all files in the repository that are larger than the specified max_file_size. The exec option of the find command appends the name of each file that matches the criteria to the .gitignore file.</p>
<p>Save the pre-commit file and make it executable by running the following command in Terminal:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">chmod</span> +x .git/hooks/pre-commit</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>With these changes, whenever you run a git commit command, Git will first execute the pre-commit hook, which will automatically add any files larger than 100 MB to the .gitignore file. This will prevent Git from tracking these files in the repository going forward.</p>
<blockquote class="blockquote">
<p>In the event that you accidentally committed a big file (&gt;100MB), you can reset to the last successful git master branch push</p>
</blockquote>
<div class="callout-warning callout callout-style-default callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Warning
</div>
</div>
<div class="callout-body-container callout-body">
<p>⚠️<strong>warning this will overwrite any changes you made after your last successful push</strong>⚠️</p>
</div>
</div>
<blockquote class="blockquote">
<p>If you still want to continue, you can un-comment the code and follow this instruction:</p>
<p>First, update all <code>origin/&lt;branch&gt;</code> refs to latest:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co">#git fetch --all</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Backup your current branch (e.g.&nbsp;<code>master</code>):</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co">#git branch backup-master</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Jump to the latest commit on <code>origin/master</code> :</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb7"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co">#git reset --hard origin/master</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</blockquote>
<hr>
</section>
</section>
<section id="running-kallisto" class="level1">
<h1>Running kallisto</h1>
<p><a href="https://pachterlab.github.io/kallisto/">kallisto</a> is a software that can be downloaded and unzipped into a <code>programs</code> directory outside of the repo.</p>
<blockquote class="blockquote">
<p>Kallisto is a pseudo-aligner, it does not need a genome to align to! It uses the sequences in the dataset to create it’s own ‘reference’</p>
</blockquote>
<p>In our case, it is already installed on <em>raven</em> (<code>/home/shared/kallisto/kallisto</code>), and can be checked by running the <code>version</code> command as below:</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/kallisto/kallisto</span> version</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<blockquote class="blockquote">
<p>If you need to download and install kallisto:</p>
<ol type="1">
<li>navigate to a <code>programs</code> directory outside of the repo</li>
<li>grab the applicable program from the <a href="https://pachterlab.github.io/kallisto/">website</a> using <code>wget</code></li>
<li>uncompress the file by navigating to the <code>programs/kallisto</code> directory and running <code>gunzip</code></li>
<li>check functionality with <code>index</code> command as above</li>
</ol>
</blockquote>
<hr>
</section>
<section id="downloading-reference" class="level1">
<h1>Downloading Reference</h1>
<p>This code grabs the Pacific oyster fasta file (rna.fna) of genes and does so ignoring the fact that <em>gannet</em> does not have a security certificate to authenticate (<code>--insecure</code>). This is usually not recommended however we know the server (i.e.&nbsp;we trust we’re not going to get a virus from this server).</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb9"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a><span class="co"># change to work in data directory</span></span>
<span id="cb9-2"><a href="#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb9-3"><a href="#cb9-3" aria-hidden="true" tabindex="-1"></a><span class="co"># download the rna.fna file to data directory from the gannet server</span></span>
<span id="cb9-4"><a href="#cb9-4" aria-hidden="true" tabindex="-1"></a><span class="ex">curl</span> <span class="at">--insecure</span> <span class="at">-O</span> https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/rna.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>In the next code chunk we create the index file which <strong>insert what the index file is for here</strong>. Creating the index file can take some time (it is 1.6GB!) This code is indexing the file <code>rna.fna</code> while also renaming it as <code>cgigas_roslin_rna.index</code>. <code>/home/shared/kallisto/kallisto</code> is the absolute path to the kallisto program from within the <em>raven</em> server, while the lines after the <code>index</code> command indicate where to get the data from (the <code>rna.fna</code> file) and where to write the file to.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="ex">/home/shared/kallisto/kallisto</span> <span class="dt">\</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>index <span class="at">-i</span> <span class="dt">\</span></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>../data/cgigas_roslin_rna.index <span class="dt">\</span></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>../data/rna.fna</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
<section id="download-sequence-reads" class="level1">
<h1>Download sequence reads</h1>
<p>The samples that we are looking at in this assignment are Pacific oyster <em>Crassostrea gigas</em>, 12 of which are ‘control’ oysters under normal conditions, and 12 of which are ‘desiccated’, shown in this table:</p>
<table class="table">
<tbody>
<tr class="odd">
<td>Sample</td>
<td>SampleID</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>D54</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>D55</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>D56</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>D57</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>D58</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>D59</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>M45</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>M46</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>M48</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>M49</td>
</tr>
<tr class="even">
<td>D-control</td>
<td>M89</td>
</tr>
<tr class="odd">
<td>D-control</td>
<td>M90</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N48</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N49</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N50</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N51</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N52</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N53</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N54</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N55</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N56</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N57</td>
</tr>
<tr class="even">
<td>D-desiccation</td>
<td>N58</td>
</tr>
<tr class="odd">
<td>D-desiccation</td>
<td>N59</td>
</tr>
</tbody>
</table>
<p>Sequence reads for these samples are on a public server at <a href="https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/nopp/" class="uri">https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/nopp/</a> or located at absolute path <code>/home/shared/8TB_HDD_01/sr320/github/nb-2023/Cgigas/data</code>.</p>
<p>This code uses the <code>--recursive</code> feature of <code>wget</code> to get all 24 files. Additionally as with <code>curl</code> above we are ignoring the fact there is not security certificate with <code>--no-check-certificate</code></p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb11"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># move to data directory</span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data </span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a><span class="co"># download fastq files to data directory </span></span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a><span class="fu">wget</span> <span class="at">--recursive</span> <span class="at">--no-parent</span> <span class="at">--no-directories</span> <span class="dt">\</span></span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>--no-check-certificate <span class="dt">\</span></span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>--accept <span class="st">'*.fastq.gz'</span> <span class="dt">\</span></span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/nopp/</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Check to make sure all 24 files were downloaded successfully</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="co"># move to data directory</span></span>
<span id="cb12-2"><a href="#cb12-2" aria-hidden="true" tabindex="-1"></a><span class="bu">cd</span> ../data</span>
<span id="cb12-3"><a href="#cb12-3" aria-hidden="true" tabindex="-1"></a><span class="co"># list all files that end with `.gz` extension, count the files in the list</span></span>
<span id="cb12-4"><a href="#cb12-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ls</span> <span class="pp">*</span>.gz <span class="kw">|</span> <span class="fu">wc</span> <span class="at">-l</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>24</code></pre>
</div>
</div>
<p>The next chunk performs the following steps:</p>
<ul>
<li>creates a subdirectory <code>kallisto_01</code> in the <code>output</code> folder using <code>mkdir</code></li>
<li>Uses the <code>find</code> utility to search for all files in the <code>../data/</code> directory that match the pattern <code>*fastq.gz</code>.</li>
<li>Uses the <code>basename</code> command to extract the base filename of each file (i.e., the filename without the directory path), and removes the suffix <code>_L001_R1_001.fastq.gz</code>.</li>
<li>Runs the kallisto <code>quant</code> command on each input file, with the following options:</li>
<li><code>-i ../data/cgigas_roslin_rna.index</code>: Use the kallisto index file located at <code>../data/cgigas_roslin_rna.index</code>.</li>
<li><code>-o ../output/kallisto_01/{}</code>: Write the output files to a directory called <code>../output/kallisto_01/</code> with a subdirectory named after the base filename of the input file (the {} is a placeholder for the base filename).</li>
<li><code>-t 40</code>: Use 40 threads for the computation.</li>
<li><code>--single -l 100 -s 10</code>: Specify that the input file contains single-end reads (–single), with an average read length of 100 (-l 100) and a standard deviation of 10 (-s 10).</li>
<li>The input file to process is specified using the {} placeholder, which is replaced by the base filename from the previous step.</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># mkdir ../output/kallisto_01</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">find</span> ../data/<span class="pp">*</span>fastq.gz <span class="dt">\</span></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a><span class="kw">|</span> <span class="fu">xargs</span> basename <span class="at">-s</span> _L001_R1_001.fastq.gz <span class="kw">|</span> <span class="fu">xargs</span> <span class="at">-I{}</span> /home/shared/kallisto/kallisto <span class="dt">\</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>quant <span class="at">-i</span> ../data/cgigas_roslin_rna.index <span class="dt">\</span></span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>-o ../output/kallisto_01/{} <span class="dt">\</span></span>
<span id="cb14-8"><a href="#cb14-8" aria-hidden="true" tabindex="-1"></a>-t 40 <span class="dt">\</span></span>
<span id="cb14-9"><a href="#cb14-9" aria-hidden="true" tabindex="-1"></a>--single <span class="at">-l</span> 100 <span class="at">-s</span> 10 ../data/{}_L001_R1_001.fastq.gz</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>This next command runs the <code>abundance_estimates_to_matrix.pl</code> script from the Trinity RNA-seq assembly software package to create a gene expression matrix from kallisto output files.</p>
<p>The specific options and arguments used in the command are as follows:</p>
<ul>
<li><code>perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl</code>: Run the abundance_estimates_to_matrix.pl script from Trinity.</li>
<li><code>--est_method kallisto</code>: Specify that the abundance estimates were generated using kallisto.</li>
<li><code>--gene_trans_map none</code>: Do not use a gene-to-transcript mapping file.</li>
<li><code>--out_prefix ../output/kallisto_01</code>: Use ../output/kallisto_01 as the output directory and prefix for the gene expression matrix file.</li>
<li><code>--name_sample_by_basedir</code>: Use the sample directory name (i.e., the final directory in the input file paths) as the sample name in the output matrix.</li>
<li>And then there are the kallisto abundance files to use as input for creating the gene expression matrix.</li>
</ul>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb15"><pre class="sourceCode bash code-with-copy"><code class="sourceCode bash"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">perl</span> /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl <span class="dt">\</span></span>
<span id="cb15-3"><a href="#cb15-3" aria-hidden="true" tabindex="-1"></a>--est_method kallisto <span class="dt">\</span></span>
<span id="cb15-4"><a href="#cb15-4" aria-hidden="true" tabindex="-1"></a>    <span class="at">--gene_trans_map</span> none <span class="dt">\</span></span>
<span id="cb15-5"><a href="#cb15-5" aria-hidden="true" tabindex="-1"></a>    <span class="at">--out_prefix</span> ../output/kallisto_01 <span class="dt">\</span></span>
<span id="cb15-6"><a href="#cb15-6" aria-hidden="true" tabindex="-1"></a>    <span class="at">--name_sample_by_basedir</span> <span class="dt">\</span></span>
<span id="cb15-7"><a href="#cb15-7" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D54_S145/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-8"><a href="#cb15-8" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D56_S136/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-9"><a href="#cb15-9" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D58_S144/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-10"><a href="#cb15-10" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M45_S140/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-11"><a href="#cb15-11" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M48_S137/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-12"><a href="#cb15-12" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M89_S138/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-13"><a href="#cb15-13" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D55_S146/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-14"><a href="#cb15-14" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D57_S143/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-15"><a href="#cb15-15" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/D59_S142/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-16"><a href="#cb15-16" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M46_S141/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-17"><a href="#cb15-17" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M49_S139/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-18"><a href="#cb15-18" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/M90_S147/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-19"><a href="#cb15-19" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N48_S194/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-20"><a href="#cb15-20" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N50_S187/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-21"><a href="#cb15-21" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N52_S184/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-22"><a href="#cb15-22" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N54_S193/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-23"><a href="#cb15-23" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N56_S192/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-24"><a href="#cb15-24" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N58_S195/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-25"><a href="#cb15-25" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N49_S185/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-26"><a href="#cb15-26" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N51_S186/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-27"><a href="#cb15-27" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N53_S188/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-28"><a href="#cb15-28" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N55_S190/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-29"><a href="#cb15-29" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N57_S191/abundance.tsv <span class="dt">\</span></span>
<span id="cb15-30"><a href="#cb15-30" aria-hidden="true" tabindex="-1"></a>    ../output/kallisto_01/N59_S189/abundance.tsv</span>
<span id="cb15-31"><a href="#cb15-31" aria-hidden="true" tabindex="-1"></a>    </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<hr>
</section>
<section id="running-deseq2" class="level1">
<h1>Running DESeq2</h1>
<p>This code performs differential expression analysis to identify deferentially expressed genes (DEGs) between a control condition and a desiccated condition in Pacific oysters.</p>
<p>First, it reads in a count matrix of isoform counts generated by <code>kallisto</code>, with row names set to the gene/transcript IDs and the first column removed. It then rounds the counts to whole numbers.</p>
<p>The <code>results()</code> function is used to extract the results table, which is ordered by gene/transcript ID.</p>
<p>The code then prints the top few rows of the results table and calculates the number of DEGs with an adjusted p-value less than or equal to 0.05. It plots the log2 fold changes versus the mean normalized counts for all genes, highlighting significant DEGs in red and adding horizontal lines at 2-fold upregulation and downregulation. Finally, it writes the list of significant DEGs to a file called “DEGlist.tab”.</p>
<p>Read in count matrix</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/kallisto_01.isoform.counts.matrix"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(countmatrix) <span class="ot">&lt;-</span> countmatrix<span class="sc">$</span>X</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> countmatrix[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(countmatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>               D54_S145 D56_S136 D58_S144 M45_S140 M48_S137 M89_S138 D55_S146
XM_011445263.3   3.0000   6.0000  3.00000  6.00000  15.0000  6.00000   5.0000
XR_004600192.1   0.0000   0.0000  5.58188  0.00000   0.0000  0.00000   0.0000
XM_034469868.1  65.9994  65.9996 48.00000 51.99950  60.9995 51.99950  68.9997
XM_034479134.1   0.0000   0.0000  0.00000  0.00000   0.0000  0.00000   0.0000
XM_034446498.1   0.0000   0.0000  0.00000  0.00000   0.0000 20.77020   0.0000
XM_034459635.1   0.0000   4.0000  1.00000  0.16843   0.0000  3.73389   0.0000
               D57_S143 D59_S142 M46_S141 M49_S139    M90_S147 N48_S194
XM_011445263.3  12.0000   6.0000   3.0000  3.00000 8.000000000   3.0000
XR_004600192.1   0.0000   0.0000   0.0000  0.00000 1.129220000   1.0000
XM_034469868.1  59.9994  42.9998  50.9997  5.31287 0.000298258  63.9995
XM_034479134.1   0.0000   0.0000   0.0000  0.00000 0.000000000   0.0000
XM_034446498.1   0.0000   0.0000   0.0000  0.00000 0.000000000   0.0000
XM_034459635.1   0.0000   0.0000   1.0000  1.00000 0.000000000   0.0000
               N50_S187 N52_S184 N54_S193 N56_S192 N58_S195 N49_S185 N51_S186
XM_011445263.3   6.0000   2.0000  4.00000        1    9.000   8.0000    2.000
XR_004600192.1   0.0000   0.0000  5.58188        0    0.000   0.0000    4.000
XM_034469868.1  74.9993  64.9997 39.99960        0  118.999  88.9993  138.998
XM_034479134.1   0.0000   0.0000  0.00000        0    0.000   0.0000    0.000
XM_034446498.1   0.0000   2.0000  0.00000        0    0.000   0.0000    0.000
XM_034459635.1   0.0000   0.0000  0.00000        0    0.000   0.0000    0.000
               N53_S188 N55_S190 N57_S191 N59_S189
XM_011445263.3   7.0000   2.0000   4.0000  6.00000
XR_004600192.1   0.0000   0.0000   0.0000  4.07503
XM_034469868.1  80.9993  59.9996  24.9999 40.99970
XM_034479134.1   0.0000   0.0000   0.0000  0.00000
XM_034446498.1   0.0000   0.0000   0.0000  0.00000
XM_034459635.1   0.0000   0.0000   1.0000  0.00000</code></pre>
</div>
</div>
<p>Round integers up to whole numbers for analysis</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> <span class="fu">round</span>(countmatrix, <span class="dv">0</span>)</span>
<span id="cb18-2"><a href="#cb18-2" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(countmatrix)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   73307 obs. of  24 variables:
 $ D54_S145: num  3 0 66 0 0 0 3 0 0 0 ...
 $ D56_S136: num  6 0 66 0 0 4 0 73 0 0 ...
 $ D58_S144: num  3 6 48 0 0 1 0 74 0 0 ...
 $ M45_S140: num  6 0 52 0 0 0 0 0 0 0 ...
 $ M48_S137: num  15 0 61 0 0 0 0 49 0 0 ...
 $ M89_S138: num  6 0 52 0 21 4 0 70 0 0 ...
 $ D55_S146: num  5 0 69 0 0 0 0 85 0 0 ...
 $ D57_S143: num  12 0 60 0 0 0 0 50 0 0 ...
 $ D59_S142: num  6 0 43 0 0 0 0 57 0 0 ...
 $ M46_S141: num  3 0 51 0 0 1 3 67 0 0 ...
 $ M49_S139: num  3 0 5 0 0 1 0 68 0 0 ...
 $ M90_S147: num  8 1 0 0 0 0 0 65 0 0 ...
 $ N48_S194: num  3 1 64 0 0 0 7 18 0 0 ...
 $ N50_S187: num  6 0 75 0 0 0 0 50 0 0 ...
 $ N52_S184: num  2 0 65 0 2 0 0 59 0 0 ...
 $ N54_S193: num  4 6 40 0 0 0 0 72 0 0 ...
 $ N56_S192: num  1 0 0 0 0 0 0 0 0 0 ...
 $ N58_S195: num  9 0 119 0 0 0 6 0 0 0 ...
 $ N49_S185: num  8 0 89 0 0 0 0 50 0 0 ...
 $ N51_S186: num  2 4 139 0 0 0 13 0 0 0 ...
 $ N53_S188: num  7 0 81 0 0 0 0 56 0 0 ...
 $ N55_S190: num  2 0 60 0 0 0 0 0 0 0 ...
 $ N57_S191: num  4 0 25 0 0 1 0 62 0 0 ...
 $ N59_S189: num  6 4 41 0 0 0 0 47 0 0 ...</code></pre>
</div>
</div>
<p>Here the code creates a <code>data.frame</code> named <code>deseq2.colData</code> containing information about the experimental conditions (control &amp; desiccated). It uses the column data dataframe named <code>deseq2.colData</code> to create a <code>DESeqDataSet</code> object using the <code>DESeqDataSetFromMatrix</code> function from the DESeq2 package.</p>
<div class="cell" data-layout-align="center" data-hash="03-dge-pretty_cache/html/unnamed-chunk-14_9fef0dda41a45bd15fdd394639dbb49c">
<div class="sourceCode cell-code" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a>deseq2.colData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">condition=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"control"</span>, <span class="dv">12</span>), <span class="fu">rep</span>(<span class="st">"desicated"</span>, <span class="dv">12</span>))), </span>
<span id="cb20-2"><a href="#cb20-2" aria-hidden="true" tabindex="-1"></a>                             <span class="at">type=</span><span class="fu">factor</span>(<span class="fu">rep</span>(<span class="st">"single-read"</span>, <span class="dv">24</span>)))</span>
<span id="cb20-3"><a href="#cb20-3" aria-hidden="true" tabindex="-1"></a><span class="co"># set row names to match the column names in the count matrix</span></span>
<span id="cb20-4"><a href="#cb20-4" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(deseq2.colData) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb20-5"><a href="#cb20-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb20-6"><a href="#cb20-6" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeqDataSet object created using the `DESeqDataSetFromMatrix` function</span></span>
<span id="cb20-7"><a href="#cb20-7" aria-hidden="true" tabindex="-1"></a>deseq2.dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> countmatrix,</span>
<span id="cb20-8"><a href="#cb20-8" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">colData =</span> deseq2.colData, </span>
<span id="cb20-9"><a href="#cb20-9" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">design =</span> <span class="sc">~</span> condition)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>converting counts to integer mode</code></pre>
</div>
</div>
<p>The <code>DESeqDataSet</code> object, named <code>deseq.dds</code>, is then passed to the <code>DESeq()</code> function to fit a negative binomial model and estimate dispersions.</p>
<div class="cell" data-layout-align="center" data-hash="03-dge-pretty_cache/html/unnamed-chunk-15_f6b149cecc3e9faf9e99693ae3a85ebf">
<div class="sourceCode cell-code" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a>deseq2.dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(deseq2.dds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating size factors</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>gene-wise dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>mean-dispersion relationship</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>final dispersion estimates</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>-- replacing outliers and refitting for 5677 genes
-- DESeq argument 'minReplicatesForReplace' = 7 
-- original counts are preserved in counts(dds)</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>estimating dispersions</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>fitting model and testing</code></pre>
</div>
<div class="sourceCode cell-code" id="cb32"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb32-1"><a href="#cb32-1" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> <span class="fu">results</span>(deseq2.dds)</span>
<span id="cb32-2"><a href="#cb32-2" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> deseq2.res[<span class="fu">order</span>(<span class="fu">rownames</span>(deseq2.res)), ]</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deseq2.res)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>log2 fold change (MLE): condition desicated vs control 
Wald test p-value: condition desicated vs control 
DataFrame with 6 rows and 6 columns
                 baseMean log2FoldChange     lfcSE      stat      pvalue
                &lt;numeric&gt;      &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt;   &lt;numeric&gt;
NM_001305288.1   0.181270      1.0453698  3.002647  0.348149 7.27728e-01
NM_001305289.1   0.881457     -2.8119577  1.068276 -2.632239 8.48240e-03
NM_001305290.1 145.913728      0.4580323  0.116185  3.942251 8.07203e-05
NM_001305291.1   0.261701      0.5618449  1.587076  0.354013 7.23329e-01
NM_001305292.1   2.902430     -1.2181330  0.763421 -1.595624 1.10573e-01
NM_001305293.1 234.342117      0.0663449  0.131969  0.502731 6.15154e-01
                     padj
                &lt;numeric&gt;
NM_001305288.1         NA
NM_001305289.1         NA
NM_001305290.1 0.00956401
NM_001305291.1         NA
NM_001305292.1 0.59541971
NM_001305293.1 0.95562321</code></pre>
</div>
</div>
<p>Count number of hits with adjusted p-value less then 0.05</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] 607   6</code></pre>
</div>
</div>
<section id="plotting-deseq2-matrix-counts" class="level3">
<h3 class="anchored" data-anchor-id="plotting-deseq2-matrix-counts">Plotting DESeq2 matrix counts</h3>
<section id="principle-component-analysis-plot" class="level4">
<h4 class="anchored" data-anchor-id="principle-component-analysis-plot">Principle Component Analysis plot</h4>
<p>What is PCA? Very simply, PCA dimensionally reduces the number of variables of a data set while preserving as much information as possible.</p>
<div class="callout-info callout callout-style-default no-icon callout-captioned">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-caption-container flex-fill">
Info
</div>
</div>
<div class="callout-body-container callout-body">
<p>Checkout <a href="https://youtu.be/FD4DeN81ODY">this YouTube video</a> that visually explains PCA!</p>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb37"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb37-1"><a href="#cb37-1" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(deseq2.dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb37-2"><a href="#cb37-2" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"condition"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="03-dge-pretty_files/figure-html/PCA-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="heatmap-plot" class="level4">
<h4 class="anchored" data-anchor-id="heatmap-plot">Heatmap plot</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top 50 differentially expressed genes</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(deseq2.dds)</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a>res_ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), ]</span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_ordered)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts and normalize</span></span>
<span id="cb38-6"><a href="#cb38-6" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(deseq2.dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb38-7"><a href="#cb38-7" aria-hidden="true" tabindex="-1"></a>counts_top <span class="ot">&lt;-</span> counts[top_genes, ]</span>
<span id="cb38-8"><a href="#cb38-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-transform counts</span></span>
<span id="cb38-9"><a href="#cb38-9" aria-hidden="true" tabindex="-1"></a>log_counts_top <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts_top <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb38-10"><a href="#cb38-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate heatmap</span></span>
<span id="cb38-11"><a href="#cb38-11" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log_counts_top, <span class="at">scale =</span> <span class="st">"row"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="03-dge-pretty_files/figure-html/heatmap-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="understanding-log2-fold-change" class="level4">
<h4 class="anchored" data-anchor-id="understanding-log2-fold-change">Understanding Log2 Fold Change</h4>
<p>First Checkout this great article <a href="https://biocorecrg.github.io/CRG_Bioinformatics_for_Biologists/differential_gene_expression.html#:~:text=Fold%20change%3A&amp;text=This%20value%20is%20typically%20reported,of%202%5E1.5%20%E2%89%88%202.82.">Comparing experimental conditions: differential expression analysis</a></p>
<p>In the article linked above, <em>Fold Change</em> is explained: Fold change: For a given comparison, a positive fold change value indicates an increase of expression, while a negative fold change indicates a decrease in expression. This value is typically reported in logarithmic scale (base 2). For example, log2 fold change of 1.5 for a specific gene in the “WT vs KO comparison” means that the expression of that gene is increased in WT relative to KO by a multiplicative factor of 2^1.5 ≈ 2.82.</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb39"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb39-1"><a href="#cb39-1" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> deseq2.res</span>
<span id="cb39-2"><a href="#cb39-2" aria-hidden="true" tabindex="-1"></a><span class="co"># The main plot</span></span>
<span id="cb39-3"><a href="#cb39-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tmp<span class="sc">$</span>baseMean, tmp<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>), <span class="at">log=</span><span class="st">"x"</span>, <span class="at">col=</span><span class="st">"darkgray"</span>,</span>
<span id="cb39-4"><a href="#cb39-4" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"DEG Dessication  (pval &lt;= 0.05)"</span>,</span>
<span id="cb39-5"><a href="#cb39-5" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"mean of normalized counts"</span>,</span>
<span id="cb39-6"><a href="#cb39-6" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"Log2 Fold Change"</span>)</span>
<span id="cb39-7"><a href="#cb39-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting the significant points and plotting them again so they're a different color</span></span>
<span id="cb39-8"><a href="#cb39-8" aria-hidden="true" tabindex="-1"></a>tmp.sig <span class="ot">&lt;-</span> deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb39-9"><a href="#cb39-9" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tmp.sig<span class="sc">$</span>baseMean, tmp.sig<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb39-10"><a href="#cb39-10" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 FC lines</span></span>
<span id="cb39-11"><a href="#cb39-11" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">col=</span><span class="st">"blue"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="volcano-plot" class="level4">
<h4 class="anchored" data-anchor-id="volcano-plot">Volcano plot</h4>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare the data for plotting</span></span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(deseq2.res)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a>res_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_df)</span>
<span id="cb40-4"><a href="#cb40-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Create volcano plot</span></span>
<span id="cb40-5"><a href="#cb40-5" aria-hidden="true" tabindex="-1"></a>volcano_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb40-6"><a href="#cb40-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb40-7"><a href="#cb40-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb40-8"><a href="#cb40-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Volcano Plot"</span>,</span>
<span id="cb40-9"><a href="#cb40-9" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb40-10"><a href="#cb40-10" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"-Log10 Adjusted P-value"</span>,</span>
<span id="cb40-11"><a href="#cb40-11" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Significantly</span><span class="sc">\n</span><span class="st">Differentially Expressed"</span>) <span class="sc">+</span></span>
<span id="cb40-12"><a href="#cb40-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb40-13"><a href="#cb40-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-14"><a href="#cb40-14" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb40-15"><a href="#cb40-15" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"top"</span>)</span>
<span id="cb40-16"><a href="#cb40-16" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcano_plot)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 41435 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="03-dge-pretty_files/figure-html/newplot-1.png" class="img-fluid figure-img" width="576"></p>
</figure>
</div>
</div>
</div>
<p>Save the list of Differentially Expressed Genes (DEGs)! Write the output to a table</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp.sig, <span class="st">"../output/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">row.names =</span> T)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Let’s take a look at this list…</p>
<div class="cell" data-layout-align="center">
<div class="sourceCode cell-code" id="cb43"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb43-1"><a href="#cb43-1" aria-hidden="true" tabindex="-1"></a>deglist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb43-2"><a href="#cb43-2" aria-hidden="true" tabindex="-1"></a>deglist<span class="sc">$</span>RowName <span class="ot">&lt;-</span> <span class="fu">rownames</span>(deglist)</span>
<span id="cb43-3"><a href="#cb43-3" aria-hidden="true" tabindex="-1"></a>deglist2 <span class="ot">&lt;-</span> deglist[, <span class="fu">c</span>(<span class="st">"RowName"</span>, <span class="st">"pvalue"</span>)] <span class="co"># Optionally, reorder the columns</span></span>
<span id="cb43-4"><a href="#cb43-4" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(deglist2)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                      RowName       pvalue
NM_001305290.1 NM_001305290.1 8.072030e-05
NM_001305310.1 NM_001305310.1 2.840948e-09
NM_001305333.1 NM_001305333.1 8.958816e-27
NM_001305339.1 NM_001305339.1 4.017360e-05
NM_001305362.1 NM_001305362.1 1.724961e-08
NM_001305373.1 NM_001305373.1 5.435026e-04
NM_001308854.1 NM_001308854.1 7.943380e-04
NM_001308861.1 NM_001308861.1 7.478144e-04
NM_001308879.1 NM_001308879.1 1.968479e-04
NM_001308913.1 NM_001308913.1 4.643630e-24
NM_001308924.1 NM_001308924.1 1.269734e-15
XM_011413623.3 XM_011413623.3 3.341473e-04
XM_011413703.3 XM_011413703.3 1.283206e-07
XM_011413735.3 XM_011413735.3 7.905600e-04
XM_011413737.3 XM_011413737.3 2.676974e-07
XM_011413957.2 XM_011413957.2 3.219768e-04
XM_011413989.3 XM_011413989.3 1.628797e-05
XM_011414589.3 XM_011414589.3 3.515308e-04
XM_011414951.3 XM_011414951.3 2.104364e-04
XM_011415190.3 XM_011415190.3 8.445948e-04
XM_011416606.3 XM_011416606.3 4.258033e-06
XM_011416726.3 XM_011416726.3 8.365440e-09
XM_011416892.3 XM_011416892.3 3.542968e-04
XM_011416960.3 XM_011416960.3 7.719401e-04
XM_011417063.3 XM_011417063.3 1.599678e-05
XM_011417064.3 XM_011417064.3 2.463779e-05
XM_011417316.3 XM_011417316.3 8.282149e-04
XM_011417317.3 XM_011417317.3 1.790747e-06
XM_011417325.3 XM_011417325.3 8.269997e-06
XM_011417799.3 XM_011417799.3 8.727681e-05
XM_011417830.3 XM_011417830.3 4.033709e-04
XM_011417902.3 XM_011417902.3 2.481092e-06
XM_011417904.3 XM_011417904.3 6.196655e-05
XM_011417908.2 XM_011417908.2 1.496858e-04
XM_011418768.3 XM_011418768.3 1.052753e-04
XM_011418872.3 XM_011418872.3 6.146797e-05
XM_011418954.3 XM_011418954.3 1.400589e-05
XM_011419041.3 XM_011419041.3 5.273378e-06
XM_011419098.3 XM_011419098.3 5.879070e-06
XM_011419179.3 XM_011419179.3 9.649033e-11
XM_011419181.3 XM_011419181.3 1.399368e-04
XM_011419576.3 XM_011419576.3 2.504552e-09
XM_011419645.3 XM_011419645.3 1.621821e-04
XM_011419848.3 XM_011419848.3 5.815604e-04
XM_011419981.3 XM_011419981.3 8.093151e-04
XM_011420357.3 XM_011420357.3 1.313937e-10
XM_011420371.3 XM_011420371.3 6.232830e-15
XM_011420372.3 XM_011420372.3 1.594096e-04
XM_011420610.3 XM_011420610.3 1.349271e-04
XM_011420631.3 XM_011420631.3 6.266213e-04
XM_011420662.3 XM_011420662.3 8.451658e-04
XM_011420824.3 XM_011420824.3 3.929574e-06
XM_011420913.3 XM_011420913.3 6.054161e-04
XM_011420959.3 XM_011420959.3 6.145874e-10
XM_011420983.3 XM_011420983.3 3.085362e-05
XM_011421012.3 XM_011421012.3 4.825696e-05
XM_011421699.3 XM_011421699.3 3.494920e-04
XM_011422329.3 XM_011422329.3 1.768261e-04
XM_011422342.3 XM_011422342.3 2.790392e-12
XM_011422632.3 XM_011422632.3 4.257736e-05
XM_011422990.3 XM_011422990.3 4.380082e-04
XM_011423572.3 XM_011423572.3 9.231433e-05
XM_011424137.3 XM_011424137.3 2.244469e-05
XM_011424142.3 XM_011424142.3 1.998279e-05
XM_011424173.3 XM_011424173.3 6.914821e-05
XM_011424596.3 XM_011424596.3 2.834022e-04
XM_011424712.3 XM_011424712.3 4.988908e-08
XM_011425176.3 XM_011425176.3 2.490140e-07
XM_011425181.3 XM_011425181.3 6.706015e-04
XM_011425228.3 XM_011425228.3 1.240413e-04
XM_011425478.3 XM_011425478.3 5.327806e-08
XM_011425483.3 XM_011425483.3 8.836866e-07
XM_011425751.3 XM_011425751.3 2.728475e-04
XM_011425772.3 XM_011425772.3 1.243186e-04
XM_011425989.3 XM_011425989.3 5.147332e-07
XM_011426213.3 XM_011426213.3 4.427574e-05
XM_011426240.3 XM_011426240.3 4.470697e-05
XM_011426321.3 XM_011426321.3 2.276575e-04
XM_011426616.3 XM_011426616.3 5.405379e-04
XM_011426647.3 XM_011426647.3 6.577317e-04
XM_011426829.3 XM_011426829.3 1.372115e-06
XM_011426844.3 XM_011426844.3 9.468090e-04
XM_011426889.3 XM_011426889.3 7.867222e-04
XM_011427084.3 XM_011427084.3 1.837089e-06
XM_011427263.3 XM_011427263.3 1.337532e-05
XM_011427409.3 XM_011427409.3 9.069434e-04
XM_011427593.3 XM_011427593.3 4.903471e-04
XM_011428050.3 XM_011428050.3 1.611654e-05
XM_011428302.3 XM_011428302.3 8.271304e-05
XM_011428362.3 XM_011428362.3 5.454487e-04
XM_011428813.3 XM_011428813.3 3.319558e-04
XM_011429405.3 XM_011429405.3 8.260076e-04
XM_011429412.3 XM_011429412.3 2.291506e-04
XM_011429469.3 XM_011429469.3 8.551627e-04
XM_011429964.3 XM_011429964.3 3.495286e-04
XM_011430087.3 XM_011430087.3 3.334114e-05
XM_011430130.2 XM_011430130.2 9.341891e-04
XM_011430224.3 XM_011430224.3 7.012039e-07
XM_011430225.3 XM_011430225.3 4.462257e-14
XM_011430243.2 XM_011430243.2 8.980812e-06
XM_011430257.3 XM_011430257.3 4.458083e-04
XM_011430301.3 XM_011430301.3 2.700597e-06
XM_011430339.3 XM_011430339.3 7.373078e-05
XM_011430436.3 XM_011430436.3 4.411238e-04
XM_011430500.3 XM_011430500.3 9.838417e-20
XM_011430508.3 XM_011430508.3 3.790199e-04
XM_011430645.3 XM_011430645.3 3.025029e-05
XM_011430696.3 XM_011430696.3 4.536149e-04
XM_011430699.3 XM_011430699.3 1.050920e-05
XM_011430921.3 XM_011430921.3 3.641187e-05
XM_011430923.3 XM_011430923.3 1.149529e-07
XM_011431087.3 XM_011431087.3 6.355809e-06
XM_011431174.3 XM_011431174.3 5.385485e-05
XM_011431272.3 XM_011431272.3 4.454120e-04
XM_011431577.3 XM_011431577.3 9.961600e-05
XM_011431655.3 XM_011431655.3 8.177482e-14
XM_011431825.3 XM_011431825.3 2.649959e-04
XM_011431987.3 XM_011431987.3 2.307602e-05
XM_011431991.3 XM_011431991.3 5.855456e-05
XM_011432386.3 XM_011432386.3 1.218110e-05
XM_011432388.3 XM_011432388.3 4.164735e-05
XM_011432500.3 XM_011432500.3 7.828421e-05
XM_011432571.3 XM_011432571.3 2.905640e-04
XM_011432647.3 XM_011432647.3 1.010575e-04
XM_011432819.3 XM_011432819.3 9.321109e-04
XM_011432843.3 XM_011432843.3 3.633636e-04
XM_011432913.3 XM_011432913.3 7.187936e-04
XM_011433143.3 XM_011433143.3 1.683032e-04
XM_011433170.3 XM_011433170.3 1.266364e-04
XM_011433250.3 XM_011433250.3 3.353189e-05
XM_011433640.3 XM_011433640.3 2.155177e-08
XM_011433641.3 XM_011433641.3 1.621288e-04
XM_011433645.3 XM_011433645.3 1.939482e-06
XM_011433685.2 XM_011433685.2 6.216901e-05
XM_011433770.3 XM_011433770.3 5.142541e-05
XM_011433782.3 XM_011433782.3 8.103463e-04
XM_011433858.3 XM_011433858.3 1.943246e-05
XM_011434034.3 XM_011434034.3 2.574077e-04
XM_011434093.3 XM_011434093.3 1.828047e-04
XM_011434186.3 XM_011434186.3 1.944177e-04
XM_011434432.3 XM_011434432.3 4.984199e-05
XM_011434464.3 XM_011434464.3 6.656349e-04
XM_011434591.3 XM_011434591.3 2.222824e-04
XM_011434867.3 XM_011434867.3 6.073406e-14
XM_011435315.3 XM_011435315.3 1.542280e-04
XM_011435569.3 XM_011435569.3 8.991256e-04
XM_011435952.3 XM_011435952.3 7.612858e-04
XM_011436004.3 XM_011436004.3 2.321961e-04
XM_011436622.3 XM_011436622.3 3.082360e-05
XM_011436623.3 XM_011436623.3 4.914438e-04
XM_011436760.3 XM_011436760.3 2.661631e-04
XM_011436951.3 XM_011436951.3 7.678790e-04
XM_011437136.3 XM_011437136.3 1.668206e-04
XM_011437363.3 XM_011437363.3 1.160009e-04
XM_011437364.3 XM_011437364.3 2.430250e-04
XM_011437548.3 XM_011437548.3 2.040561e-06
XM_011437549.3 XM_011437549.3 1.512987e-04
XM_011437582.3 XM_011437582.3 3.985999e-04
XM_011437603.3 XM_011437603.3 5.660897e-15
XM_011437870.3 XM_011437870.3 5.699066e-04
XM_011437990.3 XM_011437990.3 3.251836e-27
XM_011438030.3 XM_011438030.3 3.287878e-08
XM_011438102.3 XM_011438102.3 8.791792e-12
XM_011438148.3 XM_011438148.3 4.742145e-04
XM_011438370.3 XM_011438370.3 6.574680e-04
XM_011438573.3 XM_011438573.3 3.467842e-04
XM_011439059.3 XM_011439059.3 5.329001e-04
XM_011440038.3 XM_011440038.3 3.363217e-04
XM_011440133.3 XM_011440133.3 7.948007e-04
XM_011440434.3 XM_011440434.3 2.051335e-05
XM_011440442.3 XM_011440442.3 7.167089e-05
XM_011440790.3 XM_011440790.3 8.456193e-05
XM_011441056.3 XM_011441056.3 1.775495e-06
XM_011441100.3 XM_011441100.3 7.349289e-04
XM_011441269.3 XM_011441269.3 2.454320e-05
XM_011441367.3 XM_011441367.3 1.828695e-04
XM_011441514.3 XM_011441514.3 2.264977e-04
XM_011441523.3 XM_011441523.3 1.865142e-04
XM_011441619.3 XM_011441619.3 9.314000e-04
XM_011441686.3 XM_011441686.3 3.279884e-04
XM_011441767.3 XM_011441767.3 3.516172e-04
XM_011442016.3 XM_011442016.3 2.153606e-05
XM_011442020.3 XM_011442020.3 4.435229e-05
XM_011442175.3 XM_011442175.3 7.982896e-04
XM_011442247.3 XM_011442247.3 1.851390e-04
XM_011442617.3 XM_011442617.3 1.631372e-04
XM_011443066.3 XM_011443066.3 2.226401e-08
XM_011443186.3 XM_011443186.3 1.562053e-04
XM_011443323.3 XM_011443323.3 5.703132e-04
XM_011443599.3 XM_011443599.3 2.107523e-04
XM_011443620.3 XM_011443620.3 4.064874e-05
XM_011443836.3 XM_011443836.3 3.058180e-05
XM_011444212.3 XM_011444212.3 3.002250e-04
XM_011444215.2 XM_011444215.2 3.002250e-04
XM_011444354.3 XM_011444354.3 4.483641e-04
XM_011444450.3 XM_011444450.3 2.307010e-04
XM_011444504.3 XM_011444504.3 1.838434e-04
XM_011444981.3 XM_011444981.3 1.047025e-04
XM_011445239.3 XM_011445239.3 6.078286e-05
XM_011445240.3 XM_011445240.3 9.514162e-04
XM_011445288.3 XM_011445288.3 5.547119e-04
XM_011445572.3 XM_011445572.3 2.337850e-04
XM_011445615.3 XM_011445615.3 8.719003e-04
XM_011445887.3 XM_011445887.3 4.540377e-04
XM_011445894.3 XM_011445894.3 1.945863e-06
XM_011445901.3 XM_011445901.3 5.811564e-05
XM_011445971.3 XM_011445971.3 1.722026e-04
XM_011446090.3 XM_011446090.3 1.110955e-05
XM_011446139.3 XM_011446139.3 6.179039e-06
XM_011446277.3 XM_011446277.3 8.081690e-11
XM_011446311.3 XM_011446311.3 2.996434e-04
XM_011446726.2 XM_011446726.2 3.783771e-06
XM_011446920.3 XM_011446920.3 4.488532e-05
XM_011447253.3 XM_011447253.3 4.924094e-05
XM_011447370.3 XM_011447370.3 2.785468e-04
XM_011447398.3 XM_011447398.3 7.344561e-04
XM_011447411.3 XM_011447411.3 4.102261e-08
XM_011447654.3 XM_011447654.3 4.677597e-07
XM_011448021.3 XM_011448021.3 1.946854e-04
XM_011448507.3 XM_011448507.3 8.472484e-04
XM_011448744.3 XM_011448744.3 8.610693e-05
XM_011448822.3 XM_011448822.3 2.558331e-04
XM_011448927.3 XM_011448927.3 3.331734e-05
XM_011448982.2 XM_011448982.2 2.081174e-05
XM_011449189.3 XM_011449189.3 1.454937e-05
XM_011449346.3 XM_011449346.3 4.972052e-20
XM_011449438.3 XM_011449438.3 2.611878e-04
XM_011449497.3 XM_011449497.3 9.361044e-05
XM_011449510.3 XM_011449510.3 4.396790e-05
XM_011449723.3 XM_011449723.3 9.856303e-14
XM_011449777.3 XM_011449777.3 3.490440e-04
XM_011449847.3 XM_011449847.3 8.496448e-04
XM_011450148.3 XM_011450148.3 1.764834e-04
XM_011450198.3 XM_011450198.3 3.052080e-04
XM_011450350.3 XM_011450350.3 3.378430e-04
XM_011450391.3 XM_011450391.3 1.596057e-05
XM_011450479.3 XM_011450479.3 5.093129e-13
XM_011450557.3 XM_011450557.3 3.903654e-05
XM_011450764.3 XM_011450764.3 3.242450e-11
XM_011450795.3 XM_011450795.3 2.765626e-13
XM_011450825.3 XM_011450825.3 5.380205e-06
XM_011450883.3 XM_011450883.3 4.602595e-04
XM_011451275.3 XM_011451275.3 2.020877e-17
XM_011451289.3 XM_011451289.3 1.681236e-05
XM_011452018.3 XM_011452018.3 4.185758e-04
XM_011452055.3 XM_011452055.3 1.943938e-04
XM_011452448.3 XM_011452448.3 5.897365e-04
XM_011452951.3 XM_011452951.3 3.390612e-08
XM_011453183.3 XM_011453183.3 2.516464e-04
XM_011453270.3 XM_011453270.3 6.215093e-04
XM_011453340.3 XM_011453340.3 4.014111e-04
XM_011453448.3 XM_011453448.3 4.090423e-04
XM_011453524.3 XM_011453524.3 6.193106e-04
XM_011453578.3 XM_011453578.3 1.477097e-04
XM_011453647.3 XM_011453647.3 8.677712e-04
XM_011453904.3 XM_011453904.3 4.413058e-04
XM_011453975.3 XM_011453975.3 5.367059e-04
XM_011453979.3 XM_011453979.3 2.831819e-05
XM_011453981.3 XM_011453981.3 1.137018e-05
XM_011454256.3 XM_011454256.3 8.417929e-15
XM_011454280.3 XM_011454280.3 6.131833e-14
XM_011454878.3 XM_011454878.3 2.330149e-16
XM_011454880.3 XM_011454880.3 1.018422e-07
XM_011454889.3 XM_011454889.3 9.348774e-05
XM_011454902.3 XM_011454902.3 1.301057e-08
XM_011455053.3 XM_011455053.3 1.998647e-05
XM_011455219.3 XM_011455219.3 5.572462e-17
XM_011455306.3 XM_011455306.3 1.497907e-04
XM_011455313.3 XM_011455313.3 1.645013e-04
XM_011455565.3 XM_011455565.3 1.156866e-05
XM_011455894.3 XM_011455894.3 1.492878e-04
XM_011455935.3 XM_011455935.3 3.809274e-10
XM_011456001.3 XM_011456001.3 5.555404e-06
XM_011456072.3 XM_011456072.3 1.705088e-04
XM_011456091.3 XM_011456091.3 4.607340e-04
XM_011456107.3 XM_011456107.3 5.394168e-36
XM_011456110.3 XM_011456110.3 4.360965e-04
XM_011456111.3 XM_011456111.3 6.592538e-46
XM_011456293.3 XM_011456293.3 1.736041e-05
XM_011456384.3 XM_011456384.3 7.218239e-09
XM_011456565.3 XM_011456565.3 1.454823e-04
XM_011456757.3 XM_011456757.3 1.337984e-07
XM_011456825.3 XM_011456825.3 1.620549e-04
XM_011456828.3 XM_011456828.3 8.597025e-06
XM_011456964.3 XM_011456964.3 1.220101e-07
XM_011457389.3 XM_011457389.3 1.956801e-04
XM_011457628.3 XM_011457628.3 3.586745e-05
XM_011457656.3 XM_011457656.3 7.814105e-26
XM_011457739.3 XM_011457739.3 2.513792e-07
XM_011457784.3 XM_011457784.3 4.031792e-04
XM_011457800.3 XM_011457800.3 2.650057e-04
XM_011458560.3 XM_011458560.3 8.795141e-04
XM_011458845.3 XM_011458845.3 3.086529e-04
XM_020062944.2 XM_020062944.2 1.333335e-04
XM_020063357.2 XM_020063357.2 4.984665e-04
XM_020064309.2 XM_020064309.2 2.959270e-11
XM_020064854.2 XM_020064854.2 7.184448e-05
XM_020065965.2 XM_020065965.2 5.855552e-04
XM_020066097.2 XM_020066097.2 3.396556e-04
XM_020066098.2 XM_020066098.2 2.607384e-05
XM_020066456.2 XM_020066456.2 2.295996e-04
XM_020066648.2 XM_020066648.2 4.023964e-05
XM_020066761.2 XM_020066761.2 5.232830e-04
XM_020066967.2 XM_020066967.2 6.867529e-04
XM_020067267.2 XM_020067267.2 8.028750e-04
XM_020068045.2 XM_020068045.2 2.851443e-04
XM_020068497.2 XM_020068497.2 7.270422e-05
XM_020068971.2 XM_020068971.2 4.204261e-14
XM_020069400.2 XM_020069400.2 3.249686e-08
XM_020069537.2 XM_020069537.2 8.131657e-04
XM_020069792.2 XM_020069792.2 1.158998e-04
XM_020073627.2 XM_020073627.2 5.862084e-08
XM_020073629.2 XM_020073629.2 6.418309e-11
XM_020074026.2 XM_020074026.2 1.569009e-04
XM_020074730.2 XM_020074730.2 4.635312e-05
XM_034442975.1 XM_034442975.1 3.840336e-04
XM_034443058.1 XM_034443058.1 4.377029e-22
XM_034443149.1 XM_034443149.1 3.642549e-04
XM_034443357.1 XM_034443357.1 8.641703e-09
XM_034443414.1 XM_034443414.1 5.789018e-04
XM_034443437.1 XM_034443437.1 3.347250e-04
XM_034443632.1 XM_034443632.1 3.435105e-04
XM_034443959.1 XM_034443959.1 9.469317e-04
XM_034444076.1 XM_034444076.1 9.097028e-04
XM_034444094.1 XM_034444094.1 6.607847e-05
XM_034444206.1 XM_034444206.1 4.170721e-06
XM_034444207.1 XM_034444207.1 4.942211e-07
XM_034444279.1 XM_034444279.1 8.535144e-06
XM_034444525.1 XM_034444525.1 1.017785e-04
XM_034444541.1 XM_034444541.1 1.922887e-04
XM_034444588.1 XM_034444588.1 3.605585e-10
XM_034444608.1 XM_034444608.1 5.590865e-04
XM_034444855.1 XM_034444855.1 1.596776e-04
XM_034444927.1 XM_034444927.1 7.195018e-06
XM_034445074.1 XM_034445074.1 3.563884e-07
XM_034445114.1 XM_034445114.1 2.217731e-04
XM_034445166.1 XM_034445166.1 9.028046e-10
XM_034445168.1 XM_034445168.1 4.743123e-05
XM_034445353.1 XM_034445353.1 4.468274e-04
XM_034445394.1 XM_034445394.1 4.336090e-04
XM_034445622.1 XM_034445622.1 9.418965e-04
XM_034445729.1 XM_034445729.1 6.049445e-04
XM_034445754.1 XM_034445754.1 2.080792e-04
XM_034446150.1 XM_034446150.1 1.575587e-04
XM_034446317.1 XM_034446317.1 4.841940e-04
XM_034446360.1 XM_034446360.1 5.200532e-04
XM_034446422.1 XM_034446422.1 7.702703e-04
XM_034446803.1 XM_034446803.1 3.359864e-04
XM_034446824.1 XM_034446824.1 1.530405e-04
XM_034447263.1 XM_034447263.1 7.574925e-06
XM_034447416.1 XM_034447416.1 3.242450e-11
XM_034447633.1 XM_034447633.1 5.896243e-04
XM_034447648.1 XM_034447648.1 3.699486e-10
XM_034447649.1 XM_034447649.1 8.832120e-04
XM_034447990.1 XM_034447990.1 4.910609e-05
XM_034448172.1 XM_034448172.1 5.985733e-04
XM_034448197.1 XM_034448197.1 1.275907e-05
XM_034448205.1 XM_034448205.1 1.128786e-04
XM_034448240.1 XM_034448240.1 2.532638e-04
XM_034448598.1 XM_034448598.1 3.991274e-04
XM_034448601.1 XM_034448601.1 1.674858e-06
XM_034448605.1 XM_034448605.1 1.550253e-04
XM_034448759.1 XM_034448759.1 4.539049e-04
XM_034449458.1 XM_034449458.1 4.831280e-05
XM_034449476.1 XM_034449476.1 2.356865e-07
XM_034449510.1 XM_034449510.1 2.872963e-04
XM_034449775.1 XM_034449775.1 1.606687e-06
XM_034449992.1 XM_034449992.1 8.351628e-04
XM_034450019.1 XM_034450019.1 1.057068e-04
XM_034450047.1 XM_034450047.1 6.696095e-06
XM_034450222.1 XM_034450222.1 1.079717e-04
XM_034450223.1 XM_034450223.1 2.628245e-05
XM_034450567.1 XM_034450567.1 1.351596e-05
XM_034450895.1 XM_034450895.1 3.810627e-06
XM_034450951.1 XM_034450951.1 8.271615e-05
XM_034451123.1 XM_034451123.1 6.791529e-04
XM_034451574.1 XM_034451574.1 1.093051e-05
XM_034451583.1 XM_034451583.1 1.664446e-04
XM_034451662.1 XM_034451662.1 2.376278e-06
XM_034451680.1 XM_034451680.1 1.220911e-14
XM_034451960.1 XM_034451960.1 8.177951e-04
XM_034452505.1 XM_034452505.1 4.093442e-04
XM_034452506.1 XM_034452506.1 4.093442e-04
XM_034453221.1 XM_034453221.1 2.179040e-05
XM_034453254.1 XM_034453254.1 1.084965e-08
XM_034453255.1 XM_034453255.1 7.333308e-09
XM_034453471.1 XM_034453471.1 1.223768e-04
XM_034453633.1 XM_034453633.1 1.116710e-04
XM_034454002.1 XM_034454002.1 7.065463e-04
XM_034454255.1 XM_034454255.1 2.800494e-04
XM_034454272.1 XM_034454272.1 1.443804e-04
XM_034454297.1 XM_034454297.1 1.588872e-06
XM_034454369.1 XM_034454369.1 3.070209e-05
XM_034454536.1 XM_034454536.1 4.310799e-05
XM_034454589.1 XM_034454589.1 1.830524e-05
XM_034454728.1 XM_034454728.1 1.885604e-05
XM_034455079.1 XM_034455079.1 1.458134e-04
XM_034455182.1 XM_034455182.1 3.409669e-04
XM_034455456.1 XM_034455456.1 2.886280e-05
XM_034455724.1 XM_034455724.1 5.398412e-07
XM_034455886.1 XM_034455886.1 1.823993e-04
XM_034455914.1 XM_034455914.1 5.486992e-04
XM_034456029.1 XM_034456029.1 7.501301e-08
XM_034456207.1 XM_034456207.1 6.116171e-05
XM_034456217.1 XM_034456217.1 3.446075e-04
XM_034456345.1 XM_034456345.1 6.598070e-07
XM_034456594.1 XM_034456594.1 3.182113e-07
XM_034456595.1 XM_034456595.1 3.534841e-10
XM_034456614.1 XM_034456614.1 8.533173e-05
XM_034456624.1 XM_034456624.1 6.599626e-04
XM_034456663.1 XM_034456663.1 1.794975e-05
XM_034456876.1 XM_034456876.1 2.116838e-08
XM_034456917.1 XM_034456917.1 7.968309e-04
XM_034456921.1 XM_034456921.1 1.099714e-04
XM_034457089.1 XM_034457089.1 3.077860e-04
XM_034457172.1 XM_034457172.1 6.491191e-06
XM_034457229.1 XM_034457229.1 1.186154e-05
XM_034457255.1 XM_034457255.1 3.285643e-04
XM_034457257.1 XM_034457257.1 1.077374e-06
XM_034457808.1 XM_034457808.1 4.948001e-04
XM_034457809.1 XM_034457809.1 2.484949e-04
XM_034457811.1 XM_034457811.1 1.206279e-16
XM_034457978.1 XM_034457978.1 4.707551e-05
XM_034458218.1 XM_034458218.1 7.365589e-05
XM_034458316.1 XM_034458316.1 8.137147e-04
XM_034458350.1 XM_034458350.1 5.710488e-04
XM_034458351.1 XM_034458351.1 2.779413e-04
XM_034458358.1 XM_034458358.1 6.380094e-04
XM_034458362.1 XM_034458362.1 1.918776e-04
XM_034458699.1 XM_034458699.1 2.792520e-04
XM_034458742.1 XM_034458742.1 5.692878e-04
XM_034458842.1 XM_034458842.1 3.070360e-04
XM_034458843.1 XM_034458843.1 4.490098e-06
XM_034458844.1 XM_034458844.1 4.652436e-04
XM_034458868.1 XM_034458868.1 5.252155e-05
XM_034458912.1 XM_034458912.1 6.667182e-12
XM_034458915.1 XM_034458915.1 1.761169e-04
XM_034459041.1 XM_034459041.1 1.572705e-04
XM_034459403.1 XM_034459403.1 4.192064e-05
XM_034459707.1 XM_034459707.1 5.419411e-04
XM_034460299.1 XM_034460299.1 2.702812e-04
XM_034460679.1 XM_034460679.1 1.126577e-04
XM_034460779.1 XM_034460779.1 6.662518e-04
XM_034460988.1 XM_034460988.1 5.769239e-04
XM_034461638.1 XM_034461638.1 1.083652e-05
XM_034461819.1 XM_034461819.1 3.261584e-08
XM_034461937.1 XM_034461937.1 7.084653e-04
XM_034462019.1 XM_034462019.1 1.252199e-05
XM_034462237.1 XM_034462237.1 3.213631e-04
XM_034462308.1 XM_034462308.1 4.155848e-05
XM_034462375.1 XM_034462375.1 4.838005e-05
XM_034463330.1 XM_034463330.1 1.205410e-04
XM_034463341.1 XM_034463341.1 2.726679e-04
XM_034463907.1 XM_034463907.1 5.510531e-05
XM_034463916.1 XM_034463916.1 3.729423e-05
XM_034463962.1 XM_034463962.1 1.975056e-04
XM_034464015.1 XM_034464015.1 9.808425e-05
XM_034464143.1 XM_034464143.1 4.405850e-04
XM_034464492.1 XM_034464492.1 2.984429e-04
XM_034464576.1 XM_034464576.1 9.056514e-04
XM_034465003.1 XM_034465003.1 7.078959e-04
XM_034465520.1 XM_034465520.1 4.451425e-04
XM_034465546.1 XM_034465546.1 3.178107e-06
XM_034465703.1 XM_034465703.1 2.710304e-04
XM_034466028.1 XM_034466028.1 2.851503e-06
XM_034466255.1 XM_034466255.1 1.524822e-04
XM_034466358.1 XM_034466358.1 1.980239e-05
XM_034466383.1 XM_034466383.1 5.231909e-04
XM_034467108.1 XM_034467108.1 5.560037e-04
XM_034467233.1 XM_034467233.1 1.027358e-19
XM_034467405.1 XM_034467405.1 1.532168e-11
XM_034467444.1 XM_034467444.1 6.993862e-04
XM_034467569.1 XM_034467569.1 9.026365e-06
XM_034467743.1 XM_034467743.1 5.641001e-04
XM_034467848.1 XM_034467848.1 1.411389e-04
XM_034467900.1 XM_034467900.1 4.822431e-04
XM_034467910.1 XM_034467910.1 3.559567e-09
XM_034468024.1 XM_034468024.1 6.825914e-04
XM_034468098.1 XM_034468098.1 5.351244e-04
XM_034468207.1 XM_034468207.1 1.319507e-04
XM_034468303.1 XM_034468303.1 9.513781e-05
XM_034468386.1 XM_034468386.1 4.557490e-04
XM_034468456.1 XM_034468456.1 2.421276e-04
XM_034468483.1 XM_034468483.1 8.836776e-05
XM_034468505.1 XM_034468505.1 8.680508e-04
XM_034469181.1 XM_034469181.1 7.691647e-04
XM_034469434.1 XM_034469434.1 8.594454e-05
XM_034469641.1 XM_034469641.1 7.032130e-05
XM_034469689.1 XM_034469689.1 8.678371e-04
XM_034469778.1 XM_034469778.1 3.328354e-04
XM_034469790.1 XM_034469790.1 5.231962e-09
XM_034469941.1 XM_034469941.1 7.256329e-04
XM_034470068.1 XM_034470068.1 4.963052e-04
XM_034470240.1 XM_034470240.1 4.766704e-07
XM_034470321.1 XM_034470321.1 3.162646e-07
XM_034470338.1 XM_034470338.1 3.594938e-04
XM_034470402.1 XM_034470402.1 8.878884e-04
XM_034470406.1 XM_034470406.1 2.486472e-04
XM_034470452.1 XM_034470452.1 3.108764e-05
XM_034470781.1 XM_034470781.1 6.734960e-05
XM_034471023.1 XM_034471023.1 4.710808e-04
XM_034471029.1 XM_034471029.1 1.137404e-04
XM_034471052.1 XM_034471052.1 3.468622e-04
XM_034471087.1 XM_034471087.1 1.072786e-04
XM_034471156.1 XM_034471156.1 3.544658e-05
XM_034471171.1 XM_034471171.1 8.881152e-04
XM_034471181.1 XM_034471181.1 1.237374e-04
XM_034471468.1 XM_034471468.1 9.202984e-04
XM_034471632.1 XM_034471632.1 1.359851e-05
XM_034471694.1 XM_034471694.1 1.312102e-07
XM_034471697.1 XM_034471697.1 1.115809e-07
XM_034471759.1 XM_034471759.1 8.004160e-04
XM_034472096.1 XM_034472096.1 6.437565e-04
XM_034472105.1 XM_034472105.1 3.768369e-04
XM_034472119.1 XM_034472119.1 7.333177e-04
XM_034472167.1 XM_034472167.1 6.088103e-05
XM_034472465.1 XM_034472465.1 6.678226e-04
XM_034472524.1 XM_034472524.1 7.822116e-05
XM_034472602.1 XM_034472602.1 2.078094e-04
XM_034472902.1 XM_034472902.1 5.252285e-05
XM_034473158.1 XM_034473158.1 1.025005e-07
XM_034473208.1 XM_034473208.1 1.937385e-05
XM_034473522.1 XM_034473522.1 5.672145e-05
XM_034473839.1 XM_034473839.1 9.430323e-04
XM_034474036.1 XM_034474036.1 3.512059e-04
XM_034474325.1 XM_034474325.1 6.469764e-04
XM_034474636.1 XM_034474636.1 5.856818e-09
XM_034474649.1 XM_034474649.1 6.785902e-04
XM_034474866.1 XM_034474866.1 1.991705e-04
XM_034474911.1 XM_034474911.1 1.341414e-04
XM_034475156.1 XM_034475156.1 1.486754e-05
XM_034475157.1 XM_034475157.1 3.069844e-05
XM_034475818.1 XM_034475818.1 3.723054e-04
XM_034476264.1 XM_034476264.1 2.201291e-04
XM_034476388.1 XM_034476388.1 5.561025e-04
XM_034476392.1 XM_034476392.1 3.381783e-08
XM_034476517.1 XM_034476517.1 4.991357e-04
XM_034476550.1 XM_034476550.1 1.109685e-04
XM_034476829.1 XM_034476829.1 5.316835e-04
XM_034476955.1 XM_034476955.1 4.459265e-05
XM_034477015.1 XM_034477015.1 1.139735e-04
XM_034477591.1 XM_034477591.1 4.714748e-04
XM_034478134.1 XM_034478134.1 7.299513e-05
XM_034478177.1 XM_034478177.1 5.254694e-05
XM_034478323.1 XM_034478323.1 3.426433e-06
XM_034478793.1 XM_034478793.1 3.467088e-04
XM_034478830.1 XM_034478830.1 2.855895e-07
XM_034479290.1 XM_034479290.1 4.254355e-05
XM_034479293.1 XM_034479293.1 2.082503e-17
XM_034479294.1 XM_034479294.1 3.587471e-06
XM_034479721.1 XM_034479721.1 8.152942e-06
XM_034480188.1 XM_034480188.1 8.553143e-04
XM_034480222.1 XM_034480222.1 2.607857e-04
XM_034480331.1 XM_034480331.1 2.820922e-04
XM_034480902.1 XM_034480902.1 9.630334e-05
XM_034481530.1 XM_034481530.1 7.225446e-04
XM_034481710.1 XM_034481710.1 1.829434e-14
XM_034481737.1 XM_034481737.1 4.468638e-05
XM_034481740.1 XM_034481740.1 9.369491e-07
XM_034481838.1 XM_034481838.1 7.994673e-06
XM_034481868.1 XM_034481868.1 1.906178e-04
XM_034481890.1 XM_034481890.1 7.356342e-04
XM_034482247.1 XM_034482247.1 4.222645e-05
XM_034482265.1 XM_034482265.1 3.130388e-05
XM_034482414.1 XM_034482414.1 1.996983e-04
XM_034482422.1 XM_034482422.1 4.650988e-04
XM_034482608.1 XM_034482608.1 9.454392e-04
XM_034482887.1 XM_034482887.1 1.422445e-23
XM_034483005.1 XM_034483005.1 7.641044e-04
XM_034483031.1 XM_034483031.1 7.292612e-09
XM_034483390.1 XM_034483390.1 7.721930e-04
XM_034483436.1 XM_034483436.1 8.001592e-06
XM_034483547.1 XM_034483547.1 2.951803e-04
XM_034483593.1 XM_034483593.1 4.334156e-04
XR_004596035.1 XR_004596035.1 3.378543e-06
XR_004596351.1 XR_004596351.1 6.514039e-04
XR_004596743.1 XR_004596743.1 8.779485e-05
XR_004596906.1 XR_004596906.1 1.223076e-04
XR_004596944.1 XR_004596944.1 7.849050e-04
XR_004597797.1 XR_004597797.1 3.152069e-32
XR_004598294.1 XR_004598294.1 6.459200e-04
XR_004598549.1 XR_004598549.1 1.065098e-09
XR_004598733.1 XR_004598733.1 5.345791e-08
XR_004598744.1 XR_004598744.1 5.345791e-08
XR_004598752.1 XR_004598752.1 5.345791e-08
XR_004599693.1 XR_004599693.1 4.443712e-07
XR_004599841.1 XR_004599841.1 3.548311e-05
XR_004600141.1 XR_004600141.1 8.924312e-04
XR_004600166.1 XR_004600166.1 1.673982e-04
XR_004600498.1 XR_004600498.1 7.659371e-04
XR_004600500.1 XR_004600500.1 1.410860e-04
XR_004600655.1 XR_004600655.1 8.650703e-04
XR_004601051.1 XR_004601051.1 1.411872e-09
XR_004601053.1 XR_004601053.1 4.374660e-09
XR_004601648.1 XR_004601648.1 1.548974e-17
XR_004602054.1 XR_004602054.1 2.945989e-04
XR_004602139.1 XR_004602139.1 9.212988e-04
XR_004602422.1 XR_004602422.1 4.566694e-05
XR_004602535.1 XR_004602535.1 1.103822e-09
XR_004602542.1 XR_004602542.1 4.740261e-15
XR_004602869.1 XR_004602869.1 2.635202e-07
XR_004603302.1 XR_004603302.1 6.073043e-04
XR_004604213.1 XR_004604213.1 4.649457e-04
XR_898264.3       XR_898264.3 1.204913e-04
XR_901070.3       XR_901070.3 4.558587e-05
XR_902175.3       XR_902175.3 1.593495e-07
XR_903526.3       XR_903526.3 1.458988e-04</code></pre>
</div>
</div>
<!-- -->

</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const disableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'prefetch';
    }
  }
  const enableStylesheet = (stylesheets) => {
    for (let i=0; i < stylesheets.length; i++) {
      const stylesheet = stylesheets[i];
      stylesheet.rel = 'stylesheet';
    }
  }
  const manageTransitions = (selector, allowTransitions) => {
    const els = window.document.querySelectorAll(selector);
    for (let i=0; i < els.length; i++) {
      const el = els[i];
      if (allowTransitions) {
        el.classList.remove('notransition');
      } else {
        el.classList.add('notransition');
      }
    }
  }
  const toggleColorMode = (alternate) => {
    // Switch the stylesheets
    const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
    manageTransitions('#quarto-margin-sidebar .nav-link', false);
    if (alternate) {
      enableStylesheet(alternateStylesheets);
      for (const sheetNode of alternateStylesheets) {
        if (sheetNode.id === "quarto-bootstrap") {
          toggleBodyColorMode(sheetNode);
        }
      }
    } else {
      disableStylesheet(alternateStylesheets);
      toggleBodyColorPrimary();
    }
    manageTransitions('#quarto-margin-sidebar .nav-link', true);
    // Switch the toggles
    const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
    for (let i=0; i < toggles.length; i++) {
      const toggle = toggles[i];
      if (toggle) {
        if (alternate) {
          toggle.classList.add("alternate");     
        } else {
          toggle.classList.remove("alternate");
        }
      }
    }
    // Hack to workaround the fact that safari doesn't
    // properly recolor the scrollbar when toggling (#1455)
    if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
      manageTransitions("body", false);
      window.scrollTo(0, 1);
      setTimeout(() => {
        window.scrollTo(0, 0);
        manageTransitions("body", true);
      }, 40);  
    }
  }
  const isFileUrl = () => { 
    return window.location.protocol === 'file:';
  }
  const hasAlternateSentinel = () => {  
    let styleSentinel = getColorSchemeSentinel();
    if (styleSentinel !== null) {
      return styleSentinel === "alternate";
    } else {
      return false;
    }
  }
  const setStyleSentinel = (alternate) => {
    const value = alternate ? "alternate" : "default";
    if (!isFileUrl()) {
      window.localStorage.setItem("quarto-color-scheme", value);
    } else {
      localAlternateSentinel = value;
    }
  }
  const getColorSchemeSentinel = () => {
    if (!isFileUrl()) {
      const storageValue = window.localStorage.getItem("quarto-color-scheme");
      return storageValue != null ? storageValue : localAlternateSentinel;
    } else {
      return localAlternateSentinel;
    }
  }
  let localAlternateSentinel = 'default';
  // Dark / light mode switch
  window.quartoToggleColorScheme = () => {
    // Read the current dark / light value 
    let toAlternate = !hasAlternateSentinel();
    toggleColorMode(toAlternate);
    setStyleSentinel(toAlternate);
  };
  // Ensure there is a toggle, if there isn't float one in the top right
  if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
    const a = window.document.createElement('a');
    a.classList.add('top-right');
    a.classList.add('quarto-color-scheme-toggle');
    a.href = "";
    a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
    const i = window.document.createElement("i");
    i.classList.add('bi');
    a.appendChild(i);
    window.document.body.appendChild(a);
  }
  // Switch to dark mode if need be
  if (hasAlternateSentinel()) {
    toggleColorMode(true);
  } else {
    toggleColorMode(false);
  }
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb45" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb45-1"><a href="#cb45-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb45-2"><a href="#cb45-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Differential Gene Expression Analysis with Kallisto &amp; DESeq2"</span></span>
<span id="cb45-3"><a href="#cb45-3" aria-hidden="true" tabindex="-1"></a><span class="an">subtitle:</span><span class="co"> "a pretty report for [FISH 546](https://sr320.github.io/course-fish546-2023/assignments/03-knit.html)"</span></span>
<span id="cb45-4"><a href="#cb45-4" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Sarah Tanja"</span></span>
<span id="cb45-5"><a href="#cb45-5" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "`r format(Sys.time(), '%d %B, %Y')`"  </span></span>
<span id="cb45-6"><a href="#cb45-6" aria-hidden="true" tabindex="-1"></a><span class="an">format:</span></span>
<span id="cb45-7"><a href="#cb45-7" aria-hidden="true" tabindex="-1"></a><span class="co">  html:</span></span>
<span id="cb45-8"><a href="#cb45-8" aria-hidden="true" tabindex="-1"></a><span class="co">    code-fold: false</span></span>
<span id="cb45-9"><a href="#cb45-9" aria-hidden="true" tabindex="-1"></a><span class="co">    code-tools: true</span></span>
<span id="cb45-10"><a href="#cb45-10" aria-hidden="true" tabindex="-1"></a><span class="co">    code-copy: true</span></span>
<span id="cb45-11"><a href="#cb45-11" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight-style: github</span></span>
<span id="cb45-12"><a href="#cb45-12" aria-hidden="true" tabindex="-1"></a><span class="co">    code-overflow: wrap</span></span>
<span id="cb45-13"><a href="#cb45-13" aria-hidden="true" tabindex="-1"></a><span class="co">    theme:</span></span>
<span id="cb45-14"><a href="#cb45-14" aria-hidden="true" tabindex="-1"></a><span class="co">      light: sandstone</span></span>
<span id="cb45-15"><a href="#cb45-15" aria-hidden="true" tabindex="-1"></a><span class="co">      dark: darkly</span></span>
<span id="cb45-16"><a href="#cb45-16" aria-hidden="true" tabindex="-1"></a><span class="an">output:</span><span class="co"> </span></span>
<span id="cb45-17"><a href="#cb45-17" aria-hidden="true" tabindex="-1"></a><span class="co">  html_document:</span></span>
<span id="cb45-18"><a href="#cb45-18" aria-hidden="true" tabindex="-1"></a><span class="co">    theme: sandstone</span></span>
<span id="cb45-19"><a href="#cb45-19" aria-hidden="true" tabindex="-1"></a><span class="co">    highlight: zenburn</span></span>
<span id="cb45-20"><a href="#cb45-20" aria-hidden="true" tabindex="-1"></a><span class="co">    toc: true</span></span>
<span id="cb45-21"><a href="#cb45-21" aria-hidden="true" tabindex="-1"></a><span class="co">    toc_float: true</span></span>
<span id="cb45-22"><a href="#cb45-22" aria-hidden="true" tabindex="-1"></a><span class="co">    number_sections: true</span></span>
<span id="cb45-23"><a href="#cb45-23" aria-hidden="true" tabindex="-1"></a><span class="co">    code_folding: show</span></span>
<span id="cb45-24"><a href="#cb45-24" aria-hidden="true" tabindex="-1"></a><span class="co">    code_download: true</span></span>
<span id="cb45-25"><a href="#cb45-25" aria-hidden="true" tabindex="-1"></a><span class="co">    callout-appearance: default</span></span>
<span id="cb45-26"><a href="#cb45-26" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb45-27"><a href="#cb45-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-28"><a href="#cb45-28" aria-hidden="true" tabindex="-1"></a><span class="in">```{r setup, include=FALSE}</span></span>
<span id="cb45-29"><a href="#cb45-29" aria-hidden="true" tabindex="-1"></a>knitr<span class="sc">::</span>opts_chunk<span class="sc">$</span><span class="fu">set</span>(</span>
<span id="cb45-30"><a href="#cb45-30" aria-hidden="true" tabindex="-1"></a>  <span class="at">highlight =</span> <span class="cn">TRUE</span>,    <span class="co"># Highlight code</span></span>
<span id="cb45-31"><a href="#cb45-31" aria-hidden="true" tabindex="-1"></a>  <span class="at">echo =</span> <span class="cn">TRUE</span>,         <span class="co"># Display code chunks</span></span>
<span id="cb45-32"><a href="#cb45-32" aria-hidden="true" tabindex="-1"></a>  <span class="at">eval =</span> <span class="cn">FALSE</span>,        <span class="co"># Evaluate code chunks</span></span>
<span id="cb45-33"><a href="#cb45-33" aria-hidden="true" tabindex="-1"></a>  <span class="at">warning =</span> <span class="cn">TRUE</span>,     <span class="co"># Hide warnings</span></span>
<span id="cb45-34"><a href="#cb45-34" aria-hidden="true" tabindex="-1"></a>  <span class="at">message =</span> <span class="cn">TRUE</span>,     <span class="co"># Hide messages</span></span>
<span id="cb45-35"><a href="#cb45-35" aria-hidden="true" tabindex="-1"></a>  <span class="at">fig.width =</span> <span class="dv">6</span>,       <span class="co"># Set plot width in inches</span></span>
<span id="cb45-36"><a href="#cb45-36" aria-hidden="true" tabindex="-1"></a>  <span class="at">fig.height =</span> <span class="dv">4</span>,      <span class="co"># Set plot height in inches</span></span>
<span id="cb45-37"><a href="#cb45-37" aria-hidden="true" tabindex="-1"></a>  <span class="at">fig.align =</span> <span class="st">"center"</span> <span class="co"># Align plots to the center</span></span>
<span id="cb45-38"><a href="#cb45-38" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb45-39"><a href="#cb45-39" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-40"><a href="#cb45-40" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-41"><a href="#cb45-41" aria-hidden="true" tabindex="-1"></a><span class="fu"># Advanced Prep</span></span>
<span id="cb45-42"><a href="#cb45-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-43"><a href="#cb45-43" aria-hidden="true" tabindex="-1"></a>For this assignment you will be taking RNA-seq data, and running a common differential gene expression analysis, without the use of a reference genome. The end product will be a number of plots and table of differentially expressed genes.</span>
<span id="cb45-44"><a href="#cb45-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-45"><a href="#cb45-45" aria-hidden="true" tabindex="-1"></a>::: callout-important</span>
<span id="cb45-46"><a href="#cb45-46" aria-hidden="true" tabindex="-1"></a>**Pre-reqs:** This code contains relative paths that only work within the *raven* Roberts' Lab Server, where the data for this assignment is housed</span>
<span id="cb45-47"><a href="#cb45-47" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb45-48"><a href="#cb45-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-49"><a href="#cb45-49" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-50"><a href="#cb45-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-51"><a href="#cb45-51" aria-hidden="true" tabindex="-1"></a>Install Packages</span>
<span id="cb45-52"><a href="#cb45-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-53"><a href="#cb45-53" aria-hidden="true" tabindex="-1"></a><span class="in">```{r packages, eval=TRUE, cache=TRUE}</span></span>
<span id="cb45-54"><a href="#cb45-54" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-55"><a href="#cb45-55" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"knitr"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">"knitr"</span>) </span>
<span id="cb45-56"><a href="#cb45-56" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"tidyverse"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'tidyverse'</span>)</span>
<span id="cb45-57"><a href="#cb45-57" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"kableExtra"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'kableExtra'</span>)</span>
<span id="cb45-58"><a href="#cb45-58" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"pheatmap"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'pheatmap'</span>) </span>
<span id="cb45-59"><a href="#cb45-59" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"RColorBrewer"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'RColorBrewer'</span>) </span>
<span id="cb45-60"><a href="#cb45-60" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"data.table"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'data.table'</span>) </span>
<span id="cb45-61"><a href="#cb45-61" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"DT"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'DT'</span>)</span>
<span id="cb45-62"><a href="#cb45-62" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"Biostrings"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'Biostrings'</span>) </span>
<span id="cb45-63"><a href="#cb45-63" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"BiocManager"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) <span class="fu">install.packages</span>(<span class="st">'BiocManager'</span>) </span>
<span id="cb45-64"><a href="#cb45-64" aria-hidden="true" tabindex="-1"></a><span class="cf">if</span> (<span class="st">"DESeq2"</span> <span class="sc">%in%</span> <span class="fu">rownames</span>(<span class="fu">installed.packages</span>()) <span class="sc">==</span> <span class="st">'FALSE'</span>) BiocManager<span class="sc">::</span><span class="fu">install</span>(<span class="st">'DESeq2'</span>)</span>
<span id="cb45-65"><a href="#cb45-65" aria-hidden="true" tabindex="-1"></a> </span>
<span id="cb45-66"><a href="#cb45-66" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-67"><a href="#cb45-67" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-68"><a href="#cb45-68" aria-hidden="true" tabindex="-1"></a>Load Libraries</span>
<span id="cb45-69"><a href="#cb45-69" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-70"><a href="#cb45-70" aria-hidden="true" tabindex="-1"></a><span class="in">```{r libraries, eval=TRUE, cache=TRUE, results='hide'}</span></span>
<span id="cb45-71"><a href="#cb45-71" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb45-72"><a href="#cb45-72" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb45-73"><a href="#cb45-73" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb45-74"><a href="#cb45-74" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DESeq2)</span>
<span id="cb45-75"><a href="#cb45-75" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(pheatmap)</span>
<span id="cb45-76"><a href="#cb45-76" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(RColorBrewer)</span>
<span id="cb45-77"><a href="#cb45-77" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(data.table)</span>
<span id="cb45-78"><a href="#cb45-78" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(DT)</span>
<span id="cb45-79"><a href="#cb45-79" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(Biostrings)</span>
<span id="cb45-80"><a href="#cb45-80" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-81"><a href="#cb45-81" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-82"><a href="#cb45-82" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-83"><a href="#cb45-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-84"><a href="#cb45-84" aria-hidden="true" tabindex="-1"></a><span class="fu">## Setup git hooks to automatically gitignore large files </span></span>
<span id="cb45-85"><a href="#cb45-85" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-86"><a href="#cb45-86" aria-hidden="true" tabindex="-1"></a>In this assignment we're dealing with large files that will end up in the ../data and ../output directories. To prevent those large files from clogging up our ability to 'git push', we can use built-in hooks to automatically ignore files larger than 100 MB (no matter the directory or file name!). Here are the steps to follow:</span>
<span id="cb45-87"><a href="#cb45-87" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-88"><a href="#cb45-88" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Create a new text file in the <span class="in">`.git/hooks/`</span> directory of your repository called <span class="in">`pre-commit`</span></span>
<span id="cb45-89"><a href="#cb45-89" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-90"><a href="#cb45-90" aria-hidden="true" tabindex="-1"></a><span class="at">    &gt; *Select the `More` tab with the gear icon under the RStudio Files navigator bar and select 'show hidden files' to see the .git folder*.</span></span>
<span id="cb45-91"><a href="#cb45-91" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-92"><a href="#cb45-92" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Add the following text to the <span class="in">`.git/hooks/pre-commit`</span> file:</span>
<span id="cb45-93"><a href="#cb45-93" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-94"><a href="#cb45-94" aria-hidden="true" tabindex="-1"></a><span class="in">```         </span></span>
<span id="cb45-95"><a href="#cb45-95" aria-hidden="true" tabindex="-1"></a><span class="in">#!/bin/sh</span></span>
<span id="cb45-96"><a href="#cb45-96" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-97"><a href="#cb45-97" aria-hidden="true" tabindex="-1"></a><span class="in"># Maximum file size (in bytes)</span></span>
<span id="cb45-98"><a href="#cb45-98" aria-hidden="true" tabindex="-1"></a><span class="in">max_file_size=104857600</span></span>
<span id="cb45-99"><a href="#cb45-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-100"><a href="#cb45-100" aria-hidden="true" tabindex="-1"></a><span class="in"># Find all files larger than max_file_size and add them to the .gitignore file</span></span>
<span id="cb45-101"><a href="#cb45-101" aria-hidden="true" tabindex="-1"></a><span class="in">find . -type f -size +$max_file_size -exec echo "{}" &gt;&gt; .gitignore \;</span></span>
<span id="cb45-102"><a href="#cb45-102" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-103"><a href="#cb45-103" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-104"><a href="#cb45-104" aria-hidden="true" tabindex="-1"></a>This code sets the max_file_size variable to 100 MB and then uses the find command to locate all files in the repository that are larger than the specified max_file_size. The exec option of the find command appends the name of each file that matches the criteria to the .gitignore file.</span>
<span id="cb45-105"><a href="#cb45-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-106"><a href="#cb45-106" aria-hidden="true" tabindex="-1"></a>Save the pre-commit file and make it executable by running the following command in Terminal:</span>
<span id="cb45-107"><a href="#cb45-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-108"><a href="#cb45-108" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-109"><a href="#cb45-109" aria-hidden="true" tabindex="-1"></a>chmod <span class="sc">+</span>x .git<span class="sc">/</span>hooks<span class="sc">/</span>pre<span class="sc">-</span>commit</span>
<span id="cb45-110"><a href="#cb45-110" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-111"><a href="#cb45-111" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-112"><a href="#cb45-112" aria-hidden="true" tabindex="-1"></a>With these changes, whenever you run a git commit command, Git will first execute the pre-commit hook, which will automatically add any files larger than 100 MB to the .gitignore file. This will prevent Git from tracking these files in the repository going forward.</span>
<span id="cb45-113"><a href="#cb45-113" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-114"><a href="#cb45-114" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; In the event that you accidentally committed a big file (</span><span class="sc">\&gt;</span><span class="at">100MB), you can reset to the last successful git master branch push</span></span>
<span id="cb45-115"><a href="#cb45-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-116"><a href="#cb45-116" aria-hidden="true" tabindex="-1"></a>::: callout-warning</span>
<span id="cb45-117"><a href="#cb45-117" aria-hidden="true" tabindex="-1"></a>⚠️**warning this will overwrite any changes you made after your last successful push**⚠️</span>
<span id="cb45-118"><a href="#cb45-118" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb45-119"><a href="#cb45-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-120"><a href="#cb45-120" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; If you still want to continue, you can un-comment the code and follow this instruction:</span></span>
<span id="cb45-121"><a href="#cb45-121" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-122"><a href="#cb45-122" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; First, update all </span><span class="in">`origin/&lt;branch&gt;`</span><span class="at"> refs to latest:</span></span>
<span id="cb45-123"><a href="#cb45-123" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-124"><a href="#cb45-124" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```{r, engine='bash'}</span></span>
<span id="cb45-125"><a href="#cb45-125" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; #git fetch --all</span></span>
<span id="cb45-126"><a href="#cb45-126" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```</span></span>
<span id="cb45-127"><a href="#cb45-127" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-128"><a href="#cb45-128" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Backup your current branch (e.g. </span><span class="in">`master`</span><span class="at">):</span></span>
<span id="cb45-129"><a href="#cb45-129" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-130"><a href="#cb45-130" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```{r, engine='bash'}</span></span>
<span id="cb45-131"><a href="#cb45-131" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; #git branch backup-master</span></span>
<span id="cb45-132"><a href="#cb45-132" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```</span></span>
<span id="cb45-133"><a href="#cb45-133" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-134"><a href="#cb45-134" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Jump to the latest commit on </span><span class="in">`origin/master`</span><span class="at"> :</span></span>
<span id="cb45-135"><a href="#cb45-135" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-136"><a href="#cb45-136" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```{r, engine='bash'}</span></span>
<span id="cb45-137"><a href="#cb45-137" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; #git reset --hard origin/master</span></span>
<span id="cb45-138"><a href="#cb45-138" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```</span></span>
<span id="cb45-139"><a href="#cb45-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-140"><a href="#cb45-140" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-141"><a href="#cb45-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-142"><a href="#cb45-142" aria-hidden="true" tabindex="-1"></a><span class="fu"># Running kallisto</span></span>
<span id="cb45-143"><a href="#cb45-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-144"><a href="#cb45-144" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">kallisto</span><span class="co">](https://pachterlab.github.io/kallisto/)</span> is a software that can be downloaded and unzipped into a <span class="in">`programs`</span> directory outside of the repo.</span>
<span id="cb45-145"><a href="#cb45-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-146"><a href="#cb45-146" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; Kallisto is a pseudo-aligner, it does not need a genome to align to! It uses the sequences in the dataset to create it's own 'reference'</span></span>
<span id="cb45-147"><a href="#cb45-147" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-148"><a href="#cb45-148" aria-hidden="true" tabindex="-1"></a>In our case, it is already installed on *raven* (<span class="in">`/home/shared/kallisto/kallisto`</span>), and can be checked by running the <span class="in">`version`</span> command as below:</span>
<span id="cb45-149"><a href="#cb45-149" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-150"><a href="#cb45-150" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-151"><a href="#cb45-151" aria-hidden="true" tabindex="-1"></a><span class="sc">/</span>home<span class="sc">/</span>shared<span class="sc">/</span>kallisto<span class="sc">/</span>kallisto version</span>
<span id="cb45-152"><a href="#cb45-152" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-153"><a href="#cb45-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-154"><a href="#cb45-154" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; If you need to download and install kallisto:</span></span>
<span id="cb45-155"><a href="#cb45-155" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb45-156"><a href="#cb45-156" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; 1.  navigate to a </span><span class="in">`programs`</span><span class="at"> directory outside of the repo</span></span>
<span id="cb45-157"><a href="#cb45-157" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; 2.  grab the applicable program from the </span><span class="co">[</span><span class="ot">website</span><span class="co">](https://pachterlab.github.io/kallisto/)</span><span class="at"> using </span><span class="in">`wget`</span></span>
<span id="cb45-158"><a href="#cb45-158" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; 3.  uncompress the file by navigating to the </span><span class="in">`programs/kallisto`</span><span class="at"> directory and running </span><span class="in">`gunzip`</span></span>
<span id="cb45-159"><a href="#cb45-159" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; 4.  check functionality with </span><span class="in">`index`</span><span class="at"> command as above</span></span>
<span id="cb45-160"><a href="#cb45-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-161"><a href="#cb45-161" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-162"><a href="#cb45-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-163"><a href="#cb45-163" aria-hidden="true" tabindex="-1"></a><span class="fu"># Downloading Reference</span></span>
<span id="cb45-164"><a href="#cb45-164" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-165"><a href="#cb45-165" aria-hidden="true" tabindex="-1"></a>This code grabs the Pacific oyster fasta file (rna.fna) of genes and does so ignoring the fact that *gannet* does not have a security certificate to authenticate (<span class="in">`--insecure`</span>). This is usually not recommended however we know the server (i.e. we trust we're not going to get a virus from this server).</span>
<span id="cb45-166"><a href="#cb45-166" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-167"><a href="#cb45-167" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-168"><a href="#cb45-168" aria-hidden="true" tabindex="-1"></a><span class="co"># change to work in data directory</span></span>
<span id="cb45-169"><a href="#cb45-169" aria-hidden="true" tabindex="-1"></a>cd ..<span class="sc">/</span>data</span>
<span id="cb45-170"><a href="#cb45-170" aria-hidden="true" tabindex="-1"></a><span class="co"># download the rna.fna file to data directory from the gannet server</span></span>
<span id="cb45-171"><a href="#cb45-171" aria-hidden="true" tabindex="-1"></a>curl <span class="sc">--</span>insecure <span class="sc">-</span>O https<span class="sc">:</span><span class="er">//</span>gannet.fish.washington.edu<span class="sc">/</span>seashell<span class="sc">/</span>bu<span class="sc">-</span>github<span class="sc">/</span>nb<span class="dv">-2023</span><span class="sc">/</span>Cgigas<span class="sc">/</span>data<span class="sc">/</span>rna.fna</span>
<span id="cb45-172"><a href="#cb45-172" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-173"><a href="#cb45-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-174"><a href="#cb45-174" aria-hidden="true" tabindex="-1"></a>In the next code chunk we create the index file which **insert what the index file is for here**. Creating the index file can take some time (it is 1.6GB!) This code is indexing the file <span class="in">`rna.fna`</span> while also renaming it as <span class="in">`cgigas_roslin_rna.index`</span>. <span class="in">`/home/shared/kallisto/kallisto`</span> is the absolute path to the kallisto program from within the *raven* server, while the lines after the <span class="in">`index`</span> command indicate where to get the data from (the <span class="in">`rna.fna`</span> file) and where to write the file to.</span>
<span id="cb45-175"><a href="#cb45-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-176"><a href="#cb45-176" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-177"><a href="#cb45-177" aria-hidden="true" tabindex="-1"></a><span class="sc">/</span>home<span class="sc">/</span>shared<span class="sc">/</span>kallisto<span class="sc">/</span>kallisto \</span>
<span id="cb45-178"><a href="#cb45-178" aria-hidden="true" tabindex="-1"></a>index <span class="sc">-</span>i \</span>
<span id="cb45-179"><a href="#cb45-179" aria-hidden="true" tabindex="-1"></a>..<span class="sc">/</span>data<span class="sc">/</span>cgigas_roslin_rna.index \</span>
<span id="cb45-180"><a href="#cb45-180" aria-hidden="true" tabindex="-1"></a>..<span class="sc">/</span>data<span class="sc">/</span>rna.fna</span>
<span id="cb45-181"><a href="#cb45-181" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-182"><a href="#cb45-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-183"><a href="#cb45-183" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-184"><a href="#cb45-184" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-185"><a href="#cb45-185" aria-hidden="true" tabindex="-1"></a><span class="fu"># Download sequence reads</span></span>
<span id="cb45-186"><a href="#cb45-186" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-187"><a href="#cb45-187" aria-hidden="true" tabindex="-1"></a>The samples that we are looking at in this assignment are Pacific oyster *Crassostrea gigas*, 12 of which are 'control' oysters under normal conditions, and 12 of which are 'desiccated', shown in this table:</span>
<span id="cb45-188"><a href="#cb45-188" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-189"><a href="#cb45-189" aria-hidden="true" tabindex="-1"></a>|               |          |</span>
<span id="cb45-190"><a href="#cb45-190" aria-hidden="true" tabindex="-1"></a>|---------------|----------|</span>
<span id="cb45-191"><a href="#cb45-191" aria-hidden="true" tabindex="-1"></a>| Sample        | SampleID |</span>
<span id="cb45-192"><a href="#cb45-192" aria-hidden="true" tabindex="-1"></a>| D-control     | D54      |</span>
<span id="cb45-193"><a href="#cb45-193" aria-hidden="true" tabindex="-1"></a>| D-control     | D55      |</span>
<span id="cb45-194"><a href="#cb45-194" aria-hidden="true" tabindex="-1"></a>| D-control     | D56      |</span>
<span id="cb45-195"><a href="#cb45-195" aria-hidden="true" tabindex="-1"></a>| D-control     | D57      |</span>
<span id="cb45-196"><a href="#cb45-196" aria-hidden="true" tabindex="-1"></a>| D-control     | D58      |</span>
<span id="cb45-197"><a href="#cb45-197" aria-hidden="true" tabindex="-1"></a>| D-control     | D59      |</span>
<span id="cb45-198"><a href="#cb45-198" aria-hidden="true" tabindex="-1"></a>| D-control     | M45      |</span>
<span id="cb45-199"><a href="#cb45-199" aria-hidden="true" tabindex="-1"></a>| D-control     | M46      |</span>
<span id="cb45-200"><a href="#cb45-200" aria-hidden="true" tabindex="-1"></a>| D-control     | M48      |</span>
<span id="cb45-201"><a href="#cb45-201" aria-hidden="true" tabindex="-1"></a>| D-control     | M49      |</span>
<span id="cb45-202"><a href="#cb45-202" aria-hidden="true" tabindex="-1"></a>| D-control     | M89      |</span>
<span id="cb45-203"><a href="#cb45-203" aria-hidden="true" tabindex="-1"></a>| D-control     | M90      |</span>
<span id="cb45-204"><a href="#cb45-204" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N48      |</span>
<span id="cb45-205"><a href="#cb45-205" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N49      |</span>
<span id="cb45-206"><a href="#cb45-206" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N50      |</span>
<span id="cb45-207"><a href="#cb45-207" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N51      |</span>
<span id="cb45-208"><a href="#cb45-208" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N52      |</span>
<span id="cb45-209"><a href="#cb45-209" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N53      |</span>
<span id="cb45-210"><a href="#cb45-210" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N54      |</span>
<span id="cb45-211"><a href="#cb45-211" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N55      |</span>
<span id="cb45-212"><a href="#cb45-212" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N56      |</span>
<span id="cb45-213"><a href="#cb45-213" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N57      |</span>
<span id="cb45-214"><a href="#cb45-214" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N58      |</span>
<span id="cb45-215"><a href="#cb45-215" aria-hidden="true" tabindex="-1"></a>| D-desiccation | N59      |</span>
<span id="cb45-216"><a href="#cb45-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-217"><a href="#cb45-217" aria-hidden="true" tabindex="-1"></a>Sequence reads for these samples are on a public server at <span class="ot">&lt;https://gannet.fish.washington.edu/seashell/bu-github/nb-2023/Cgigas/data/nopp/&gt;</span> or located at absolute path <span class="in">`/home/shared/8TB_HDD_01/sr320/github/nb-2023/Cgigas/data`</span>.</span>
<span id="cb45-218"><a href="#cb45-218" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-219"><a href="#cb45-219" aria-hidden="true" tabindex="-1"></a>This code uses the <span class="in">`--recursive`</span> feature of <span class="in">`wget`</span> to get all 24 files. Additionally as with <span class="in">`curl`</span> above we are ignoring the fact there is not security certificate with <span class="in">`--no-check-certificate`</span></span>
<span id="cb45-220"><a href="#cb45-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-221"><a href="#cb45-221" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-222"><a href="#cb45-222" aria-hidden="true" tabindex="-1"></a><span class="co"># move to data directory</span></span>
<span id="cb45-223"><a href="#cb45-223" aria-hidden="true" tabindex="-1"></a>cd ..<span class="sc">/</span>data </span>
<span id="cb45-224"><a href="#cb45-224" aria-hidden="true" tabindex="-1"></a><span class="co"># download fastq files to data directory </span></span>
<span id="cb45-225"><a href="#cb45-225" aria-hidden="true" tabindex="-1"></a>wget <span class="sc">--</span>recursive <span class="sc">--</span>no<span class="sc">-</span>parent <span class="sc">--</span>no<span class="sc">-</span>directories \</span>
<span id="cb45-226"><a href="#cb45-226" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>no<span class="sc">-</span>check<span class="sc">-</span>certificate \</span>
<span id="cb45-227"><a href="#cb45-227" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>accept <span class="st">'*.fastq.gz'</span> \</span>
<span id="cb45-228"><a href="#cb45-228" aria-hidden="true" tabindex="-1"></a>https<span class="sc">:</span><span class="er">//</span>gannet.fish.washington.edu<span class="sc">/</span>seashell<span class="sc">/</span>bu<span class="sc">-</span>github<span class="sc">/</span>nb<span class="dv">-2023</span><span class="sc">/</span>Cgigas<span class="sc">/</span>data<span class="sc">/</span>nopp<span class="sc">/</span></span>
<span id="cb45-229"><a href="#cb45-229" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-230"><a href="#cb45-230" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-231"><a href="#cb45-231" aria-hidden="true" tabindex="-1"></a>Check to make sure all 24 files were downloaded successfully</span>
<span id="cb45-232"><a href="#cb45-232" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-233"><a href="#cb45-233" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash', eval=TRUE}</span></span>
<span id="cb45-234"><a href="#cb45-234" aria-hidden="true" tabindex="-1"></a><span class="co"># move to data directory</span></span>
<span id="cb45-235"><a href="#cb45-235" aria-hidden="true" tabindex="-1"></a>cd ..<span class="sc">/</span>data</span>
<span id="cb45-236"><a href="#cb45-236" aria-hidden="true" tabindex="-1"></a><span class="co"># list all files that end with `.gz` extension, count the files in the list</span></span>
<span id="cb45-237"><a href="#cb45-237" aria-hidden="true" tabindex="-1"></a>ls <span class="sc">*</span>.gz <span class="sc">|</span> wc <span class="sc">-</span>l</span>
<span id="cb45-238"><a href="#cb45-238" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-239"><a href="#cb45-239" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-240"><a href="#cb45-240" aria-hidden="true" tabindex="-1"></a>The next chunk performs the following steps:</span>
<span id="cb45-241"><a href="#cb45-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-242"><a href="#cb45-242" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>creates a subdirectory <span class="in">`kallisto_01`</span> in the <span class="in">`output`</span> folder using <span class="in">`mkdir`</span></span>
<span id="cb45-243"><a href="#cb45-243" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Uses the <span class="in">`find`</span> utility to search for all files in the <span class="in">`../data/`</span> directory that match the pattern <span class="in">`*fastq.gz`</span>.</span>
<span id="cb45-244"><a href="#cb45-244" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Uses the <span class="in">`basename`</span> command to extract the base filename of each file (i.e., the filename without the directory path), and removes the suffix <span class="in">`_L001_R1_001.fastq.gz`</span>.</span>
<span id="cb45-245"><a href="#cb45-245" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>Runs the kallisto <span class="in">`quant`</span> command on each input file, with the following options:</span>
<span id="cb45-246"><a href="#cb45-246" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`-i ../data/cgigas_roslin_rna.index`</span>: Use the kallisto index file located at <span class="in">`../data/cgigas_roslin_rna.index`</span>.</span>
<span id="cb45-247"><a href="#cb45-247" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`-o ../output/kallisto_01/{}`</span>: Write the output files to a directory called <span class="in">`../output/kallisto_01/`</span> with a subdirectory named after the base filename of the input file (the {} is a placeholder for the base filename).</span>
<span id="cb45-248"><a href="#cb45-248" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`-t 40`</span>: Use 40 threads for the computation.</span>
<span id="cb45-249"><a href="#cb45-249" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`--single -l 100 -s 10`</span>: Specify that the input file contains single-end reads (--single), with an average read length of 100 (-l 100) and a standard deviation of 10 (-s 10).</span>
<span id="cb45-250"><a href="#cb45-250" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>The input file to process is specified using the {} placeholder, which is replaced by the base filename from the previous step.</span>
<span id="cb45-251"><a href="#cb45-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-252"><a href="#cb45-252" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-253"><a href="#cb45-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-254"><a href="#cb45-254" aria-hidden="true" tabindex="-1"></a><span class="co"># mkdir ../output/kallisto_01</span></span>
<span id="cb45-255"><a href="#cb45-255" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-256"><a href="#cb45-256" aria-hidden="true" tabindex="-1"></a>find ..<span class="sc">/</span>data<span class="sc">/</span><span class="er">*</span>fastq.gz \</span>
<span id="cb45-257"><a href="#cb45-257" aria-hidden="true" tabindex="-1"></a><span class="sc">|</span> xargs basename <span class="sc">-</span>s _L001_R1_001.fastq.gz <span class="sc">|</span> xargs <span class="sc">-</span>I{} <span class="sc">/</span>home<span class="sc">/</span>shared<span class="sc">/</span>kallisto<span class="sc">/</span>kallisto \</span>
<span id="cb45-258"><a href="#cb45-258" aria-hidden="true" tabindex="-1"></a>quant <span class="sc">-</span>i ..<span class="sc">/</span>data<span class="sc">/</span>cgigas_roslin_rna.index \</span>
<span id="cb45-259"><a href="#cb45-259" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span>o ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>{} \</span>
<span id="cb45-260"><a href="#cb45-260" aria-hidden="true" tabindex="-1"></a><span class="sc">-</span>t <span class="dv">40</span> \</span>
<span id="cb45-261"><a href="#cb45-261" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>single <span class="sc">-</span>l <span class="dv">100</span> <span class="sc">-</span>s <span class="dv">10</span> ..<span class="sc">/</span>data<span class="sc">/</span>{}_L001_R1_001.fastq.gz</span>
<span id="cb45-262"><a href="#cb45-262" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-263"><a href="#cb45-263" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-264"><a href="#cb45-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-265"><a href="#cb45-265" aria-hidden="true" tabindex="-1"></a>This next command runs the <span class="in">`abundance_estimates_to_matrix.pl`</span> script from the Trinity RNA-seq assembly software package to create a gene expression matrix from kallisto output files.</span>
<span id="cb45-266"><a href="#cb45-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-267"><a href="#cb45-267" aria-hidden="true" tabindex="-1"></a>The specific options and arguments used in the command are as follows:</span>
<span id="cb45-268"><a href="#cb45-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-269"><a href="#cb45-269" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`perl /home/shared/trinityrnaseq-v2.12.0/util/abundance_estimates_to_matrix.pl`</span>: Run the abundance_estimates_to_matrix.pl script from Trinity.</span>
<span id="cb45-270"><a href="#cb45-270" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`--est_method kallisto`</span>: Specify that the abundance estimates were generated using kallisto.</span>
<span id="cb45-271"><a href="#cb45-271" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`--gene_trans_map none`</span>: Do not use a gene-to-transcript mapping file.</span>
<span id="cb45-272"><a href="#cb45-272" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`--out_prefix ../output/kallisto_01`</span>: Use ../output/kallisto_01 as the output directory and prefix for the gene expression matrix file.</span>
<span id="cb45-273"><a href="#cb45-273" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span><span class="in">`--name_sample_by_basedir`</span>: Use the sample directory name (i.e., the final directory in the input file paths) as the sample name in the output matrix.</span>
<span id="cb45-274"><a href="#cb45-274" aria-hidden="true" tabindex="-1"></a><span class="ss">-   </span>And then there are the kallisto abundance files to use as input for creating the gene expression matrix.</span>
<span id="cb45-275"><a href="#cb45-275" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-276"><a href="#cb45-276" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, engine='bash'}</span></span>
<span id="cb45-277"><a href="#cb45-277" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-278"><a href="#cb45-278" aria-hidden="true" tabindex="-1"></a>perl <span class="sc">/</span>home<span class="sc">/</span>shared<span class="sc">/</span>trinityrnaseq<span class="sc">-</span>v2.<span class="fl">12.0</span><span class="sc">/</span>util<span class="sc">/</span>abundance_estimates_to_matrix.pl \</span>
<span id="cb45-279"><a href="#cb45-279" aria-hidden="true" tabindex="-1"></a><span class="sc">--</span>est_method kallisto \</span>
<span id="cb45-280"><a href="#cb45-280" aria-hidden="true" tabindex="-1"></a>    <span class="sc">--</span>gene_trans_map none \</span>
<span id="cb45-281"><a href="#cb45-281" aria-hidden="true" tabindex="-1"></a>    <span class="sc">--</span>out_prefix ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01 \</span>
<span id="cb45-282"><a href="#cb45-282" aria-hidden="true" tabindex="-1"></a>    <span class="sc">--</span>name_sample_by_basedir \</span>
<span id="cb45-283"><a href="#cb45-283" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D54_S145<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-284"><a href="#cb45-284" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D56_S136<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-285"><a href="#cb45-285" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D58_S144<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-286"><a href="#cb45-286" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M45_S140<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-287"><a href="#cb45-287" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M48_S137<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-288"><a href="#cb45-288" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M89_S138<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-289"><a href="#cb45-289" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D55_S146<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-290"><a href="#cb45-290" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D57_S143<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-291"><a href="#cb45-291" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>D59_S142<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-292"><a href="#cb45-292" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M46_S141<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-293"><a href="#cb45-293" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M49_S139<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-294"><a href="#cb45-294" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>M90_S147<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-295"><a href="#cb45-295" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N48_S194<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-296"><a href="#cb45-296" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N50_S187<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-297"><a href="#cb45-297" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N52_S184<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-298"><a href="#cb45-298" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N54_S193<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-299"><a href="#cb45-299" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N56_S192<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-300"><a href="#cb45-300" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N58_S195<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-301"><a href="#cb45-301" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N49_S185<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-302"><a href="#cb45-302" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N51_S186<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-303"><a href="#cb45-303" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N53_S188<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-304"><a href="#cb45-304" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N55_S190<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-305"><a href="#cb45-305" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N57_S191<span class="sc">/</span>abundance.tsv \</span>
<span id="cb45-306"><a href="#cb45-306" aria-hidden="true" tabindex="-1"></a>    ..<span class="sc">/</span>output<span class="sc">/</span>kallisto_01<span class="sc">/</span>N59_S189<span class="sc">/</span>abundance.tsv</span>
<span id="cb45-307"><a href="#cb45-307" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb45-308"><a href="#cb45-308" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-309"><a href="#cb45-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-310"><a href="#cb45-310" aria-hidden="true" tabindex="-1"></a>------------------------------------------------------------------------</span>
<span id="cb45-311"><a href="#cb45-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-312"><a href="#cb45-312" aria-hidden="true" tabindex="-1"></a><span class="fu"># Running DESeq2</span></span>
<span id="cb45-313"><a href="#cb45-313" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-314"><a href="#cb45-314" aria-hidden="true" tabindex="-1"></a>This code performs differential expression analysis to identify deferentially expressed genes (DEGs) between a control condition and a desiccated condition in Pacific oysters.</span>
<span id="cb45-315"><a href="#cb45-315" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-316"><a href="#cb45-316" aria-hidden="true" tabindex="-1"></a>First, it reads in a count matrix of isoform counts generated by <span class="in">`kallisto`</span>, with row names set to the gene/transcript IDs and the first column removed. It then rounds the counts to whole numbers.</span>
<span id="cb45-317"><a href="#cb45-317" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-318"><a href="#cb45-318" aria-hidden="true" tabindex="-1"></a>The <span class="in">`results()`</span> function is used to extract the results table, which is ordered by gene/transcript ID.</span>
<span id="cb45-319"><a href="#cb45-319" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-320"><a href="#cb45-320" aria-hidden="true" tabindex="-1"></a>The code then prints the top few rows of the results table and calculates the number of DEGs with an adjusted p-value less than or equal to 0.05. It plots the log2 fold changes versus the mean normalized counts for all genes, highlighting significant DEGs in red and adding horizontal lines at 2-fold upregulation and downregulation. Finally, it writes the list of significant DEGs to a file called "DEGlist.tab".</span>
<span id="cb45-321"><a href="#cb45-321" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-322"><a href="#cb45-322" aria-hidden="true" tabindex="-1"></a>Read in count matrix</span>
<span id="cb45-323"><a href="#cb45-323" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-324"><a href="#cb45-324" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb45-325"><a href="#cb45-325" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-326"><a href="#cb45-326" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> <span class="fu">read.delim</span>(<span class="st">"../output/kallisto_01.isoform.counts.matrix"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>)</span>
<span id="cb45-327"><a href="#cb45-327" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(countmatrix) <span class="ot">&lt;-</span> countmatrix<span class="sc">$</span>X</span>
<span id="cb45-328"><a href="#cb45-328" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> countmatrix[,<span class="sc">-</span><span class="dv">1</span>]</span>
<span id="cb45-329"><a href="#cb45-329" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(countmatrix)</span>
<span id="cb45-330"><a href="#cb45-330" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-331"><a href="#cb45-331" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-332"><a href="#cb45-332" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-333"><a href="#cb45-333" aria-hidden="true" tabindex="-1"></a>Round integers up to whole numbers for analysis</span>
<span id="cb45-334"><a href="#cb45-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-335"><a href="#cb45-335" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb45-336"><a href="#cb45-336" aria-hidden="true" tabindex="-1"></a>countmatrix <span class="ot">&lt;-</span> <span class="fu">round</span>(countmatrix, <span class="dv">0</span>)</span>
<span id="cb45-337"><a href="#cb45-337" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(countmatrix)</span>
<span id="cb45-338"><a href="#cb45-338" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-339"><a href="#cb45-339" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-340"><a href="#cb45-340" aria-hidden="true" tabindex="-1"></a>Here the code creates a <span class="in">`data.frame`</span> named <span class="in">`deseq2.colData`</span> containing information about the experimental conditions (control &amp; desiccated). It uses the column data dataframe named <span class="in">`deseq2.colData`</span> to create a <span class="in">`DESeqDataSet`</span> object using the <span class="in">`DESeqDataSetFromMatrix`</span> function from the DESeq2 package.</span>
<span id="cb45-341"><a href="#cb45-341" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-342"><a href="#cb45-342" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE, cache=TRUE}</span></span>
<span id="cb45-343"><a href="#cb45-343" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-344"><a href="#cb45-344" aria-hidden="true" tabindex="-1"></a>deseq2.colData <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(<span class="at">condition=</span><span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"control"</span>, <span class="dv">12</span>), <span class="fu">rep</span>(<span class="st">"desicated"</span>, <span class="dv">12</span>))), </span>
<span id="cb45-345"><a href="#cb45-345" aria-hidden="true" tabindex="-1"></a>                             <span class="at">type=</span><span class="fu">factor</span>(<span class="fu">rep</span>(<span class="st">"single-read"</span>, <span class="dv">24</span>)))</span>
<span id="cb45-346"><a href="#cb45-346" aria-hidden="true" tabindex="-1"></a><span class="co"># set row names to match the column names in the count matrix</span></span>
<span id="cb45-347"><a href="#cb45-347" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(deseq2.colData) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(data)</span>
<span id="cb45-348"><a href="#cb45-348" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-349"><a href="#cb45-349" aria-hidden="true" tabindex="-1"></a><span class="co"># DESeqDataSet object created using the `DESeqDataSetFromMatrix` function</span></span>
<span id="cb45-350"><a href="#cb45-350" aria-hidden="true" tabindex="-1"></a>deseq2.dds <span class="ot">&lt;-</span> <span class="fu">DESeqDataSetFromMatrix</span>(<span class="at">countData =</span> countmatrix,</span>
<span id="cb45-351"><a href="#cb45-351" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">colData =</span> deseq2.colData, </span>
<span id="cb45-352"><a href="#cb45-352" aria-hidden="true" tabindex="-1"></a>                                     <span class="at">design =</span> <span class="sc">~</span> condition)</span>
<span id="cb45-353"><a href="#cb45-353" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-354"><a href="#cb45-354" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-355"><a href="#cb45-355" aria-hidden="true" tabindex="-1"></a>The <span class="in">`DESeqDataSet`</span> object, named <span class="in">`deseq.dds`</span>, is then passed to the <span class="in">`DESeq()`</span> function to fit a negative binomial model and estimate dispersions.</span>
<span id="cb45-356"><a href="#cb45-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-357"><a href="#cb45-357" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE, cache=TRUE}</span></span>
<span id="cb45-358"><a href="#cb45-358" aria-hidden="true" tabindex="-1"></a>deseq2.dds <span class="ot">&lt;-</span> <span class="fu">DESeq</span>(deseq2.dds)</span>
<span id="cb45-359"><a href="#cb45-359" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> <span class="fu">results</span>(deseq2.dds)</span>
<span id="cb45-360"><a href="#cb45-360" aria-hidden="true" tabindex="-1"></a>deseq2.res <span class="ot">&lt;-</span> deseq2.res[<span class="fu">order</span>(<span class="fu">rownames</span>(deseq2.res)), ]</span>
<span id="cb45-361"><a href="#cb45-361" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-362"><a href="#cb45-362" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-363"><a href="#cb45-363" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb45-364"><a href="#cb45-364" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(deseq2.res)</span>
<span id="cb45-365"><a href="#cb45-365" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-366"><a href="#cb45-366" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-367"><a href="#cb45-367" aria-hidden="true" tabindex="-1"></a>Count number of hits with adjusted p-value less then 0.05</span>
<span id="cb45-368"><a href="#cb45-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-369"><a href="#cb45-369" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb45-370"><a href="#cb45-370" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ])</span>
<span id="cb45-371"><a href="#cb45-371" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-372"><a href="#cb45-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-373"><a href="#cb45-373" aria-hidden="true" tabindex="-1"></a><span class="fu">### Plotting DESeq2 matrix counts</span></span>
<span id="cb45-374"><a href="#cb45-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-375"><a href="#cb45-375" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Principle Component Analysis plot</span></span>
<span id="cb45-376"><a href="#cb45-376" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-377"><a href="#cb45-377" aria-hidden="true" tabindex="-1"></a>What is PCA? Very simply, PCA dimensionally reduces the number of variables of a data set while preserving as much information as possible.</span>
<span id="cb45-378"><a href="#cb45-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-379"><a href="#cb45-379" aria-hidden="true" tabindex="-1"></a>::: callout-info</span>
<span id="cb45-380"><a href="#cb45-380" aria-hidden="true" tabindex="-1"></a>Checkout <span class="co">[</span><span class="ot">this YouTube video</span><span class="co">](https://youtu.be/FD4DeN81ODY)</span> that visually explains PCA!</span>
<span id="cb45-381"><a href="#cb45-381" aria-hidden="true" tabindex="-1"></a>:::</span>
<span id="cb45-382"><a href="#cb45-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-383"><a href="#cb45-383" aria-hidden="true" tabindex="-1"></a><span class="in">```{r PCA, eval=TRUE}</span></span>
<span id="cb45-384"><a href="#cb45-384" aria-hidden="true" tabindex="-1"></a>vsd <span class="ot">&lt;-</span> <span class="fu">vst</span>(deseq2.dds, <span class="at">blind =</span> <span class="cn">FALSE</span>)</span>
<span id="cb45-385"><a href="#cb45-385" aria-hidden="true" tabindex="-1"></a><span class="fu">plotPCA</span>(vsd, <span class="at">intgroup =</span> <span class="st">"condition"</span>)</span>
<span id="cb45-386"><a href="#cb45-386" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-387"><a href="#cb45-387" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-388"><a href="#cb45-388" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Heatmap plot</span></span>
<span id="cb45-389"><a href="#cb45-389" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-390"><a href="#cb45-390" aria-hidden="true" tabindex="-1"></a><span class="in">```{r heatmap, eval=TRUE}</span></span>
<span id="cb45-391"><a href="#cb45-391" aria-hidden="true" tabindex="-1"></a><span class="co"># Select top 50 differentially expressed genes</span></span>
<span id="cb45-392"><a href="#cb45-392" aria-hidden="true" tabindex="-1"></a>res <span class="ot">&lt;-</span> <span class="fu">results</span>(deseq2.dds)</span>
<span id="cb45-393"><a href="#cb45-393" aria-hidden="true" tabindex="-1"></a>res_ordered <span class="ot">&lt;-</span> res[<span class="fu">order</span>(res<span class="sc">$</span>padj), ]</span>
<span id="cb45-394"><a href="#cb45-394" aria-hidden="true" tabindex="-1"></a>top_genes <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_ordered)[<span class="dv">1</span><span class="sc">:</span><span class="dv">50</span>]</span>
<span id="cb45-395"><a href="#cb45-395" aria-hidden="true" tabindex="-1"></a><span class="co"># Extract counts and normalize</span></span>
<span id="cb45-396"><a href="#cb45-396" aria-hidden="true" tabindex="-1"></a>counts <span class="ot">&lt;-</span> <span class="fu">counts</span>(deseq2.dds, <span class="at">normalized =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-397"><a href="#cb45-397" aria-hidden="true" tabindex="-1"></a>counts_top <span class="ot">&lt;-</span> counts[top_genes, ]</span>
<span id="cb45-398"><a href="#cb45-398" aria-hidden="true" tabindex="-1"></a><span class="co"># Log-transform counts</span></span>
<span id="cb45-399"><a href="#cb45-399" aria-hidden="true" tabindex="-1"></a>log_counts_top <span class="ot">&lt;-</span> <span class="fu">log2</span>(counts_top <span class="sc">+</span> <span class="dv">1</span>)</span>
<span id="cb45-400"><a href="#cb45-400" aria-hidden="true" tabindex="-1"></a><span class="co"># Generate heatmap</span></span>
<span id="cb45-401"><a href="#cb45-401" aria-hidden="true" tabindex="-1"></a><span class="fu">pheatmap</span>(log_counts_top, <span class="at">scale =</span> <span class="st">"row"</span>)</span>
<span id="cb45-402"><a href="#cb45-402" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-403"><a href="#cb45-403" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-404"><a href="#cb45-404" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Understanding Log2 Fold Change</span></span>
<span id="cb45-405"><a href="#cb45-405" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-406"><a href="#cb45-406" aria-hidden="true" tabindex="-1"></a>First Checkout this great article <span class="co">[</span><span class="ot">Comparing experimental conditions: differential expression analysis</span><span class="co">](https://biocorecrg.github.io/CRG_Bioinformatics_for_Biologists/differential_gene_expression.html#:~:text=Fold%20change%3A&amp;text=This%20value%20is%20typically%20reported,of%202%5E1.5%20%E2%89%88%202.82.)</span></span>
<span id="cb45-407"><a href="#cb45-407" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-408"><a href="#cb45-408" aria-hidden="true" tabindex="-1"></a>In the article linked above, *Fold Change* is explained: Fold change: For a given comparison, a positive fold change value indicates an increase of expression, while a negative fold change indicates a decrease in expression. This value is typically reported in logarithmic scale (base 2). For example, log2 fold change of 1.5 for a specific gene in the "WT vs KO comparison" means that the expression of that gene is increased in WT relative to KO by a multiplicative factor of 2\^1.5 ≈ 2.82.</span>
<span id="cb45-409"><a href="#cb45-409" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-410"><a href="#cb45-410" aria-hidden="true" tabindex="-1"></a><span class="in">```{r results='markup', warning=FALSE}</span></span>
<span id="cb45-411"><a href="#cb45-411" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> deseq2.res</span>
<span id="cb45-412"><a href="#cb45-412" aria-hidden="true" tabindex="-1"></a><span class="co"># The main plot</span></span>
<span id="cb45-413"><a href="#cb45-413" aria-hidden="true" tabindex="-1"></a><span class="fu">plot</span>(tmp<span class="sc">$</span>baseMean, tmp<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">ylim=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">3</span>, <span class="dv">3</span>), <span class="at">log=</span><span class="st">"x"</span>, <span class="at">col=</span><span class="st">"darkgray"</span>,</span>
<span id="cb45-414"><a href="#cb45-414" aria-hidden="true" tabindex="-1"></a>     <span class="at">main=</span><span class="st">"DEG Dessication  (pval &lt;= 0.05)"</span>,</span>
<span id="cb45-415"><a href="#cb45-415" aria-hidden="true" tabindex="-1"></a>     <span class="at">xlab=</span><span class="st">"mean of normalized counts"</span>,</span>
<span id="cb45-416"><a href="#cb45-416" aria-hidden="true" tabindex="-1"></a>     <span class="at">ylab=</span><span class="st">"Log2 Fold Change"</span>)</span>
<span id="cb45-417"><a href="#cb45-417" aria-hidden="true" tabindex="-1"></a><span class="co"># Getting the significant points and plotting them again so they're a different color</span></span>
<span id="cb45-418"><a href="#cb45-418" aria-hidden="true" tabindex="-1"></a>tmp.sig <span class="ot">&lt;-</span> deseq2.res[<span class="sc">!</span><span class="fu">is.na</span>(deseq2.res<span class="sc">$</span>padj) <span class="sc">&amp;</span> deseq2.res<span class="sc">$</span>padj <span class="sc">&lt;=</span> <span class="fl">0.05</span>, ]</span>
<span id="cb45-419"><a href="#cb45-419" aria-hidden="true" tabindex="-1"></a><span class="fu">points</span>(tmp.sig<span class="sc">$</span>baseMean, tmp.sig<span class="sc">$</span>log2FoldChange, <span class="at">pch=</span><span class="dv">20</span>, <span class="at">cex=</span><span class="fl">0.45</span>, <span class="at">col=</span><span class="st">"red"</span>)</span>
<span id="cb45-420"><a href="#cb45-420" aria-hidden="true" tabindex="-1"></a><span class="co"># 2 FC lines</span></span>
<span id="cb45-421"><a href="#cb45-421" aria-hidden="true" tabindex="-1"></a><span class="fu">abline</span>(<span class="at">h=</span><span class="fu">c</span>(<span class="sc">-</span><span class="dv">1</span>,<span class="dv">1</span>), <span class="at">col=</span><span class="st">"blue"</span>)</span>
<span id="cb45-422"><a href="#cb45-422" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-423"><a href="#cb45-423" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-424"><a href="#cb45-424" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Volcano plot</span></span>
<span id="cb45-425"><a href="#cb45-425" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-426"><a href="#cb45-426" aria-hidden="true" tabindex="-1"></a><span class="in">```{r newplot, eval=TRUE}</span></span>
<span id="cb45-427"><a href="#cb45-427" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare the data for plotting</span></span>
<span id="cb45-428"><a href="#cb45-428" aria-hidden="true" tabindex="-1"></a>res_df <span class="ot">&lt;-</span> <span class="fu">as.data.frame</span>(deseq2.res)</span>
<span id="cb45-429"><a href="#cb45-429" aria-hidden="true" tabindex="-1"></a>res_df<span class="sc">$</span>gene <span class="ot">&lt;-</span> <span class="fu">row.names</span>(res_df)</span>
<span id="cb45-430"><a href="#cb45-430" aria-hidden="true" tabindex="-1"></a><span class="co"># Create volcano plot</span></span>
<span id="cb45-431"><a href="#cb45-431" aria-hidden="true" tabindex="-1"></a>volcano_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(res_df, <span class="fu">aes</span>(<span class="at">x =</span> log2FoldChange, <span class="at">y =</span> <span class="sc">-</span><span class="fu">log10</span>(padj), <span class="at">color =</span> padj <span class="sc">&lt;</span> <span class="fl">0.05</span>)) <span class="sc">+</span></span>
<span id="cb45-432"><a href="#cb45-432" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="at">alpha =</span> <span class="fl">0.6</span>, <span class="at">size =</span> <span class="fl">1.5</span>) <span class="sc">+</span></span>
<span id="cb45-433"><a href="#cb45-433" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_color_manual</span>(<span class="at">values =</span> <span class="fu">c</span>(<span class="st">"grey"</span>, <span class="st">"red"</span>)) <span class="sc">+</span></span>
<span id="cb45-434"><a href="#cb45-434" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Volcano Plot"</span>,</span>
<span id="cb45-435"><a href="#cb45-435" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Log2 Fold Change"</span>,</span>
<span id="cb45-436"><a href="#cb45-436" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"-Log10 Adjusted P-value"</span>,</span>
<span id="cb45-437"><a href="#cb45-437" aria-hidden="true" tabindex="-1"></a>       <span class="at">color =</span> <span class="st">"Significantly</span><span class="sc">\n</span><span class="st">Differentially Expressed"</span>) <span class="sc">+</span></span>
<span id="cb45-438"><a href="#cb45-438" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb45-439"><a href="#cb45-439" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">panel.grid.major =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-440"><a href="#cb45-440" aria-hidden="true" tabindex="-1"></a>        <span class="at">panel.grid.minor =</span> <span class="fu">element_blank</span>(),</span>
<span id="cb45-441"><a href="#cb45-441" aria-hidden="true" tabindex="-1"></a>        <span class="at">legend.position =</span> <span class="st">"top"</span>)</span>
<span id="cb45-442"><a href="#cb45-442" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(volcano_plot)</span>
<span id="cb45-443"><a href="#cb45-443" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-444"><a href="#cb45-444" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-445"><a href="#cb45-445" aria-hidden="true" tabindex="-1"></a>Save the list of Differentially Expressed Genes (DEGs)! Write the output to a table</span>
<span id="cb45-446"><a href="#cb45-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-447"><a href="#cb45-447" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval = FALSE}</span></span>
<span id="cb45-448"><a href="#cb45-448" aria-hidden="true" tabindex="-1"></a><span class="fu">write.table</span>(tmp.sig, <span class="st">"../output/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">row.names =</span> T)</span>
<span id="cb45-449"><a href="#cb45-449" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb45-450"><a href="#cb45-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-451"><a href="#cb45-451" aria-hidden="true" tabindex="-1"></a>Let's take a look at this list...</span>
<span id="cb45-452"><a href="#cb45-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb45-453"><a href="#cb45-453" aria-hidden="true" tabindex="-1"></a><span class="in">```{r, eval=TRUE}</span></span>
<span id="cb45-454"><a href="#cb45-454" aria-hidden="true" tabindex="-1"></a>deglist <span class="ot">&lt;-</span> <span class="fu">read.csv</span>(<span class="st">"../output/DEGlist.tab"</span>, <span class="at">sep =</span> <span class="st">'</span><span class="sc">\t</span><span class="st">'</span>, <span class="at">header =</span> <span class="cn">TRUE</span>)</span>
<span id="cb45-455"><a href="#cb45-455" aria-hidden="true" tabindex="-1"></a>deglist<span class="sc">$</span>RowName <span class="ot">&lt;-</span> <span class="fu">rownames</span>(deglist)</span>
<span id="cb45-456"><a href="#cb45-456" aria-hidden="true" tabindex="-1"></a>deglist2 <span class="ot">&lt;-</span> deglist[, <span class="fu">c</span>(<span class="st">"RowName"</span>, <span class="st">"pvalue"</span>)] <span class="co"># Optionally, reorder the columns</span></span>
<span id="cb45-457"><a href="#cb45-457" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(deglist2)</span>
<span id="cb45-458"><a href="#cb45-458" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->



</body></html>